---
title: "MRG Manuscript Figs"
author: "Nathan D Olson"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
editor_options: 
  markdown: 
    wrap: 72
---

# TODO

- [ ] Sanger confirmation plot\
- [ ] Variant size distribution plot\ 
- [ ] Workout panel drafts\
- [ ] Re-org panels for figure construction\
- [ ] Construct figures\
- [ ] Generate list of supplemental data files

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
```

## High Level Workflow

**Figure**: Diagram of benchmark set development process

::: {style="width: 640px; height: 480px; margin: 10px; position: relative;"}
<iframe frameborder="0" style="width:640px; height:480px" src="https://lucid.app/documents/embeddedchart/5d688f72-e442-4f27-ae57-79bd3f6a7560" id="sh_rd3t24.Fl">

</iframe>
:::



## Supplemental Table HG002 v4.2.1 and hifiasm v0.11 Overlaps

[TODO] get file



## Selection Of Difficult Medical Genes

{DRAFT} A. Gene List Venn Diagram: `analysis/gene_list_analysis.Rmd`
*Med Genes List UpSet*

B. (Potentially cut/ move) UpSet plot med benchmark gene source  *Maybe add to gene source upset.*

{DRAFT} C. V4 overlap: `analysis/gene_list_analysis.Rmd` *Medically
Relevant Genes V4 Coverage*

{DRAFT} D. Characterization of 396 genes with \< 90% coverage. - Difficult strat coverage



## *CUT* Supplemental Figure Excluded Gene Characterization 
Incorporated into Selection of Difficult Med Genes 



## Supplemental Figure HG002 hifiasm v0.11 Assembly Quality Metrics

<https://www.google.com/url?q=https://drive.google.com/file/d/15o-2q152S206JDS92iVuRKWL4AmoB5QX/view?usp%3Dsharing&sa=D&source=editors&ust=1618002287325000&usg=AOvVaw3YVU7lPdGCwFNlXECB5eqf>

## Medical Genes Benchmark

{DRAFT} A. Gene assembly coverage - flanking diagram  - Illustrator diagram
{DRAFT}B. Medical composition - Upset plot with testing modalities - `analysis/med_genes_characterization.Rmd}
{DRAFT} C. Small variant benchmark (poster figure) over GATK   - `analysis/med_genes_characterization.Rmd` "## MRG Small Var Benchmark GATK"
*TODO* D. Sanger confirmation?   
{DRAFT} E. Coverage by gene part (exon, introns,
flanking)   - `analysis/med_genes_characterization.Rmd` does not include flanking region coverage
*TODO* F. small var sv and indel size distributions



## Table Small Variant Benchmark Statistics



## Table SV Benchmark Statistics



## (Potential To Add) Prominent things in new benchmark

Overlap with Medical Genes benchmark figure

## Figure Improvements after masking false duplications

JM Figure [TODO] Get input data and code from Jenny

## Supp Fig Masking Minimally affects whole genome accuracy

JM Figure [TODO] Get input data and code from Jenny

## Supplementary Fig SMN12 Dotplot and Graph

page 11 Dotplots and graph

## Fig SMN1 IGV

Potentially combine with SMN12 supp

## Figure RHCE

## Supplementary Figure RHD

## Figure GPI 17kb INS

IGV screenshot

## Fig GTF2IRD2 INS in SegDup

IGV screenshot

## Supplementary Figure Large Duplication

IGV screenshot

## Supplemental Figure Complex Variant TR

IGV screenshot

## Figure Manual Curation Results (Panels for Small vars and SVs??)
{DRAFT} `analysis/manual_curation_fig.Rmd`

## Figure MRG_Benchmark_Perfomance_vs_v4.2.1
{DRAFT} `analysis/med_genes_characterization.Rmd` "## MRG Small Var Benchmark GATK"

## Supplemental Figure CR1

Dotplot

## Supplemental Figure KIR

Dotplot

## Supplemental Figure PRSS1

## Supplemental Figure DUX4

## Supplementary Figure Reference_Gap_FAM20C

IGV screenshot

# Figs to add to list

1.  Supplementary Figure CBS_GRCh37: IGV screenshot
2.  Supplementary Figure NCF1: IGV screenshot
3.  Supplementary Figure DSPP: IGV screenshot
4.  Supplementary Figure TNNI3: IGV screenshot
5.  Supplementary Figure FLG: doplots

# Session Information

## System Information

```{r}
sessioninfo::platform_info()
```

## Package Versions

```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```
