---
title: "Making Tidy Dataset for MRG Manuscript"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(readxl)
library(here)
```

# Objective
Generate tidy data sets for generating manuscript figures and tables.

## Datasets
Datasets described in `data/data_descriptor_tidy_data.md`.


# Making Tidy Data Tables

## Loading MRG Benchmark Gene Lists
```{r}
## Gene List Files
mrg_gene_beds <- list(
  GRCh37 = here(
    "data",
    "gene_coords",
    "unsorted",
    "GRCh37_mrg_bench_gene.bed"
  ),
  GRCh38 = here(
    "data",
    "gene_coords",
    "unsorted",
    "GRCh38_mrg_bench_gene.bed"
  )
)

mrg_genes_df <- mrg_gene_beds %>%
  map_dfr(
    read_tsv,
    col_names = c("chrom", "start", "end", "gene"),
    col_types = "cddc",
    .id = "ref"
  ) %>% 
  mutate(in_mrg = TRUE)
```

## Gene Coordinates

Gene Body and Gene plus Flank and Segdup Coordinates
```{r}
## Full CMRG Gene list gene body coordinates
gene_df <-
    list(
        GRCh37 = here("workflow","data", "gene_coords", "GRCh37_mrg_full_gene.bed"),
        GRCh38 = here("workflow","data", "gene_coords", "GRCh38_mrg_full_gene.bed")
    ) %>%
    map_dfr(
        read_tsv,
        col_names = c("chrom", "start", "end", "gene"),
        col_types = "ciic",
        .id = "ref"
    )

## Full CMRG Gene list gene body + flank including full overlapping segdups
gene_plus_flank_df <- list(
        GRCh37 = here("workflow","data", "gene_coords", "GRCh37_mrg_full_gene-plus-flank.bed"),
        GRCh38 = here("workflow","data", "gene_coords", "GRCh38_mrg_full_gene-plus-flank.bed")
    ) %>%
    map_dfr(
        read_tsv,
        col_names = c("chrom", "start_flank", "end_flank"),
        col_types = "cii",
        .id = "ref"
    )
```

```{r}
gene_coords_df <- bind_cols(gene_df, gene_plus_flank_df)
```

__Data Checks__: Making sure the col bind command worked as expected and gene statistics are approriately assigned. Assume row numbers are 0 for all checks.
```{r}
gene_coords_df %>% 
    filter(ref...1 != ref...6)
```
```{r}
gene_coords_df %>% 
    filter(chrom...2 != chrom...7)
```
```{r}
gene_coords_df %>% 
    mutate(start_diff = start - start_flank,
           end_diff = end_flank - end) %>% 
    filter(start_diff < 0)
```
```{r}
gene_coords_df %>% 
    mutate(start_diff = start - start_flank,
           end_diff = end_flank - end) %>% 
    filter(start_diff < 0)
```

__Cleaning up data frame__
```{r}
gene_coords_df <- bind_cols(gene_df, gene_plus_flank_df) %>% 
    rename(ref = ref...1, chrom = chrom...2) %>% 
    mutate(coords = str_c(chrom, ":", start, "-", end)) %>% 
    select(gene, ref, coords, start_flank, end_flank) 
    

glimpse(gene_coords_df)
```

### Writing To File 
```{r}
write_tsv(gene_coords_df,
          here("data","tidy","gene_coords.tsv.gz"))
```



## Medical Gene List
Using Gene Lists in `data/mrg_lists` directory to generate medically relevant genes list table.
```{r}
gene_list_files <-
  list(mandelker     = "Mandelker_Medically_Relevant_Genes.tsv",
       cosmic        = "COSMIC_Gene_Census.tsv",
       high_priority = "Steve_Lincoln_Compiled_Medical_Gene_List.tsv") %>%
  map(~ here("data", "mrg_lists", .))

mrg_source_tbl <- gene_list_files %>% 
  map_dfr(read_tsv, col_types = "c", .id = "gene_list") %>% 
  add_column(in_list = "yes") %>% 
  rename(gene = Gene) %>% 
  pivot_wider(names_from = gene_list,
                values_from = in_list,
                values_fill = "no")
```




Information on testing modality and high priority gene list from Lincoln https://drive.google.com/drive/u/1/folders/1gUjMx24CkZCbt4vC46W3nQY_3DcjP4X9 


Gene testing modality from individual list generated by Justin Wagner in collaboration with Steve Lincoln.
```{r}
testing_modality_df <- list(ACMG = "ACMG_SF_2.0.tsv",
                             ACOG = "ACOG.tsv",
                             HTD = "Assorted_HTD.tsv",
                             ClinGen = "ClinGen_moderate_definitive_strong.tsv",
                             Counsyl = "Counsyl.tsv",
                             CPIC = "CPIC.tsv",
                             NCCN = "NCCN_ESMO.tsv") %>% 
    map(~here("data","mrg_lists","testing_modalities", .)) %>% 
    map_df(read_tsv, col_types = "c", .id = "testing_modality") %>% 
    count(Gene, testing_modality) %>% 
    pivot_wider(names_from = testing_modality,
                values_from = n,
                values_fill = 0) %>% 
    rename(gene = "Gene")
```

```{r}
mrg_info_tbl <- mrg_source_tbl %>%
    full_join(testing_modality_df)
```

_DATA CHECK:_ Checking that all genes in testing modality list are in mrg source list
All genes in the testing modality list are in the combined GRCh37 and GRCh38 gene lists. 

```{r}
nrow(mrg_info_tbl) == nrow(mrg_source_tbl)
```


Tidying tibble up for export
```{r}
tidy_mrg_info_tbl <- mrg_info_tbl %>%
    ## replacing NAs from testing modality with 0s
    mutate(across(where(is.integer), replace_na, 0))
```

### Example Gene
```{r}
tidy_mrg_info_tbl %>%
  filter(gene == "OR4F5") %>%
  DT::datatable()
```

### Writing To File 
```{r}
write_tsv(tidy_mrg_info_tbl,
          here("data","tidy","gene_list_info.tsv.gz"))
```


## Included Bases
```{r}
## Included Bases for Gene Body and Introns
gene_cov_files <- list.files(path = here("workflow", "results", "bench_cov_tbls"),
                             pattern = "HG002.*_gene_cov.tsv") %>%
  set_names(~ str_remove(., "_cov.tsv")) %>%
  map(~ here("workflow", "results", "bench_cov_tbls", .))

gene_cov_df <- gene_cov_files %>%
  map_dfr(
    read_tsv,
    col_names = c(
      "chrom",
      "start",
      "end",
      "gene",
      "n_regions",
      "bp_cov",
      "gene_size",
      "cov"
    ),
    col_types = "ciiciiid",
    .id = "covset"
  ) %>%
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  mutate(coords = str_c(chrom, ":", start, "-", end))

## Gene Coordinates
gene_coords_df <- gene_cov_df %>%
  select(gene, ref, chrom, coords) %>%
  distinct()

```

```{r}
glimpse(gene_cov_df)
```



```{r}
## Included Exon Bases
exon_cov_files <-
  list.files(path = here("workflow", "results", "bench_cov_tbls"),
             pattern = "HG002.*_exon_cov.tsv",
  ) %>%
  set_names( ~ str_remove(., "_cov.tsv")) %>%
  map( ~ here("workflow", "results", "bench_cov_tbls", .))

exon_cov_df <- exon_cov_files %>%
  map_dfr(
    read_tsv,
    ## ensembl colnames are assumed
    col_names = c(
      "chrom",
      "start",
      "end",
      "gene",
      "ensembl_gene_id",
      "ensembl_exon_id",
      "n_regions",
      "bp_cov",
      "exon_size",
      "frac_cov"
    ),
    col_types = "ciiccciiid",
    .id = "covset"
  ) %>%
  group_by(covset, chrom, gene, ensembl_gene_id) %>%
  summarise(total_bp_cov = sum(bp_cov),
            total_exon_size = sum(exon_size)) %>%
  mutate(cov = total_bp_cov / total_exon_size) %>%
  rename(bp_cov = total_bp_cov) %>% 
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  left_join(gene_coords_df)
```

```{r}
gene_coords_df %>% filter(str_detect(gene, "SEP"))
```

```{r}
exon_cov_df <- exon_cov_files %>%
  map_dfr(
    read_tsv,
    ## ensembl colnames are assumed
    col_names = c(
      "chrom",
      "start",
      "end",
      "gene",
      "ensembl_gene_id",
      "ensembl_exon_id",
      "n_regions",
      "bp_cov",
      "exon_size",
      "frac_cov"
    ),
    col_types = "ciiccciiid",
    .id = "covset"
  ) %>%
  group_by(covset, chrom, gene, ensembl_gene_id) %>%
  summarise(total_bp_cov = sum(bp_cov),
            total_exon_size = sum(exon_size)) %>%
  mutate(cov = total_bp_cov / total_exon_size) %>%
  rename(bp_cov = total_bp_cov) %>% 
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  left_join(gene_coords_df)
```


```{r}
glimpse(exon_cov_df)
```

```{r}
## Included Bases for Introns
intron_cov_files <-
  list.files(path = here("workflow", "results", "bench_cov_tbls"),
             pattern = "HG002.*_intron_cov.tsv",
  ) %>%
  set_names( ~ str_remove(., "_cov.tsv")) %>%
  map( ~ here("workflow", "results", "bench_cov_tbls", .))

cnames <- c("chrom", "start", "end", "gene", "n_regions", 
            "bp_cov", "intron_size", "cov")

intron_cov_df <- intron_cov_files %>%
  map_dfr(read_tsv,
          col_names = cnames,
          col_types = "ciiciiid",
          .id = "covset") %>%
  group_by(covset, chrom, gene) %>%
  summarise(total_bp_cov = sum(bp_cov),
            total_intron_size = sum(intron_size)) %>%
  mutate(cov = total_bp_cov / total_intron_size) %>%
  rename(bp_cov = total_bp_cov) %>% 
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  left_join(gene_coords_df)
```

```{r}
glimpse(intron_cov_df)
```

__QUESTION__ Different number of rows - assuming due to lack of coverage by some genes 
```{r}
nrow(exon_cov_df)
```

```{r}
nrow(gene_cov_df)
```

```{r}
nrow(intron_cov_df)
```

```{r}
## Combined Cov DF making data frames consistent with DF defined below
bench_cov_df <- bind_rows(gene_cov_df, exon_cov_df, intron_cov_df) %>%
  ungroup() %>%
  select(gene, ref, coords, benchmark, bench_type, region, cov, bp_cov) %>%
  ## Annotating with in MRG info
  left_join(mrg_genes_df) %>%
  mutate(in_mrg = if_else(is.na(in_mrg), FALSE, in_mrg)) %>% 
  select(-chrom,-start,-end)
```

```{r}
glimpse(bench_cov_df)
```


_DATA CHECK:_ Single entry per benchmark set and region coverage type. Should return 0 rows
```{r}
bench_cov_df  %>% 
  arrange(gene) %>% 
  group_by(gene, ref, benchmark, bench_type, region) %>% 
  mutate(n = n()) %>% 
  filter(n != 1)
```


_DATA CHECK:_ All genes should have coordinates. Should return 0 rows
```{r}
bench_cov_df %>% 
  filter(is.na(coords))
```

### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
bench_cov_df %>% 
    filter(gene == "AGL") %>% 
    DT::datatable(rownames = FALSE)
```

### Writing To File 
```{r}
bench_cov_file <-  here("data","tidy","bench_coverage.tsv.gz")

write_tsv(bench_cov_df, bench_cov_file)
```


## Included Difficult Bases
Copied and edited from included bases above
```{r}
## Included Bases for Gene Body and Introns
gene_diff_files <- list.files(path = here("workflow", "results", "bench_diff_bases_tbls"),
                             pattern = "HG002.*_mrg-gene-diff_cov.tsv") %>%
  set_names(~ str_remove(., "_cov.tsv")) %>%
  map(~ here("workflow", "results", "bench_diff_bases_tbls", .))

gene_diff_bases_df <- gene_diff_files %>%
  map_dfr(
    read_tsv,
    col_names = c(
      "chrom",
      "start",
      "end",
      "gene",
      "n_regions",
      "bp_cov",
      "gene_size",
      "cov"
    ),
    col_types = "ciiciiid",
    .id = "covset"
  ) %>%
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  mutate(coords = str_c(chrom, ":", start, "-", end))

## Gene Coordinates
gene_coords_df <- gene_diff_bases_df %>%
  select(gene, ref, chrom, coords) %>%
  distinct()

```

```{r}
glimpse(gene_diff_bases_df)
```



```{r}
## Included Exon Bases
exon_diff_files <-
  list.files(path = here("workflow", "results", "bench_diff_bases_tbls"),
             pattern = "HG002.*_mrg-exon-diff_cov.tsv",
  ) %>%
  set_names( ~ str_remove(., "_cov.tsv")) %>%
  map( ~ here("workflow", "results", "bench_diff_bases_tbls", .))

exon_diff_bases_df <- exon_diff_files %>%
  map_dfr(
    read_tsv,
    ## ensembl colnames are assumed
    col_names = c(
      "chrom",
      "start",
      "end",
      "gene",
      "ensembl_gene_id",
      "ensembl_exon_id",
      "n_regions",
      "bp_cov",
      "exon_size",
      "frac_cov"
    ),
    col_types = "ciiccciiid",
    .id = "covset"
  ) %>%
  group_by(covset, chrom, gene, ensembl_gene_id) %>%
  summarise(total_bp_cov = sum(bp_cov),
            total_exon_size = sum(exon_size)) %>%
  mutate(cov = total_bp_cov / total_exon_size) %>%
  rename(bp_cov = total_bp_cov) %>% 
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  left_join(gene_coords_df)
```

```{r}
glimpse(exon_diff_bases_df)
```

```{r}
## Included Bases for Introns
intron_diff_files <-
  list.files(path = here("workflow", "results", "bench_diff_bases_tbls"),
             pattern = "HG002.*_mrg-intron-diff_cov.tsv",
  ) %>%
  set_names( ~ str_remove(., "_cov.tsv")) %>%
  map( ~ here("workflow", "results", "bench_diff_bases_tbls", .))

cnames <- c("chrom", "start", "end", "gene", "n_regions", 
            "bp_cov", "intron_size", "cov")

intron_diff_bases_df <- intron_diff_files %>%
  map_dfr(read_tsv,
          col_names = cnames,
          col_types = "ciiciiid",
          .id = "covset") %>%
  group_by(covset, chrom, gene) %>%
  summarise(total_bp_cov = sum(bp_cov),
            total_intron_size = sum(intron_size)) %>%
  mutate(cov = total_bp_cov / total_intron_size) %>%
  rename(bp_cov = total_bp_cov) %>% 
  separate(
    col = covset,
    into = c("giab_id", "ref", "benchmark", "bench_type", "region"),
    sep = "_",
    remove = TRUE
  ) %>%
  left_join(gene_coords_df)
```

```{r}
glimpse(intron_diff_bases_df)
```



```{r}
## Combined Cov DF making data frames consistent with DF defined below
diff_bases_df <- bind_rows(gene_diff_bases_df, exon_diff_bases_df, intron_diff_bases_df) %>%
  ungroup() %>%
  select(gene, ref, coords, benchmark, bench_type, region, cov, bp_cov)
```

```{r}
glimpse(diff_bases_df)
```


_DATA CHECK:_ Single entry per benchmark set and region coverage type. Should return 0 rows
```{r}
diff_bases_df  %>% 
  arrange(gene) %>% 
  group_by(gene, ref, benchmark, bench_type, region) %>% 
  mutate(n = n()) %>% 
  filter(n != 1)
```


_DATA CHECK:_ All genes should have coordinates. Should return 0 rows
```{r}
diff_bases_df %>% 
  filter(is.na(coords))
```



### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
diff_bases_df %>% 
    filter(gene == "AGL") %>% 
    DT::datatable(rownames = FALSE)
```

### Writing To File 
```{r}
bench_diff_file <-  here("data","tidy","bench_diff_bases.tsv.gz")

write_tsv(diff_bases_df, bench_diff_file)
```

## Annotated Variants Tables

```{r}
## Calculating INDEL Variant Size
get_var_size <- function(REF, ALT){
  ref_bp <- nchar(REF)
  
  ## Dealing with multiple ALTs
  # --- using largest SV
  if(str_detect(ALT, ",")){
    alts <- unlist(str_split(ALT, ","))
    alts_bp <- nchar(alts)
    var_sizes <- alts_bp - ref_bp
    
    var_size <- var_sizes[abs(var_sizes) == max(abs(var_sizes))]
    
    ## returning first entry value if the two genotypes are the same size
    return(var_size[1])
  }
  return(nchar(ALT) - ref_bp)
}

## Reading annotated var vcf
read_anno_tsv <- function(anno_tsv_file) {
  ## Colnames and types for small variants (smv) and SVs (stv)
  if (str_detect(anno_tsv_file, "_smallvar_")) {
    var_type <- "smallvar"
    cnames <-
      c("CHROM", "POS", "GT", "TYPE", "GENE", "EXON", "REF", "ALT")
    ctypes <-  "cicccccc"
    
  } else if (str_detect(anno_tsv_file, "_SV_")) {
    var_type <- "SV"
    cnames <-
      c(
        "CHROM",
        "POS",
        "GT",
        "TYPE",
        "REPTYPE",
        "BREAKSIMLENGTH",
        "GENE",
        "EXON",
        "REF",
        "ALT"
      )
    ctypes <-  "cicccicccc"
  } else {
    warning("variant type not determined from file name")
    return(tibble())
  }
  
  
  
  ## Parsers for different benchmark sets
  if (str_detect(anno_tsv_file, "_cmrg_")) {
    var_df <-
      read_tsv(anno_tsv_file,
               col_names = cnames,
               col_types = ctypes)
    
    ## Getting variant size
    sized_var_df <- var_df %>%
      rowwise() %>%
      mutate(var_size = get_var_size(REF, ALT)) %>%
      select(-REF, -ALT)
    
    if (var_type == "smallvar") {
      sized_var_df <- sized_var_df %>%
        filter(abs(var_size) < 50)
    }
  } else if (str_detect(anno_tsv_file, "_v4_")) {
    sized_var_df <- read_tsv(anno_tsv_file,
                             col_names = cnames,
                             col_types = ctypes) %>%
      ## Excluding variants not in mrg gene list
      filter(GENE != ".") %>%
      rowwise() %>%
      mutate(var_size = get_var_size(REF, ALT)) %>%
      select(-REF,-ALT) %>%
      filter(abs(var_size) < 50)
    
  } else if (str_detect(anno_tsv_file, "_v0.6_")) {
    sized_var_df <- read_tsv(anno_tsv_file,
                             col_names = cnames,
                             col_types = ctypes) %>%
      ## Excluding variants not in mrg gene list and non-HG002 variants
      filter(GENE != ".",
             GT %in% c("0/1", "1/1")) %>%
      rowwise() %>%
      mutate(var_size = get_var_size(REF, ALT)) %>%
      select(-REF,-ALT)
  } else{
    warning("benchmark set not determined from file name")
    return(tibble())
  }
  
  ## Returning size variant table
  return(sized_var_df)
}

## Generating data frame from individual annotated vcf tables
var_tbl_files <- list.files(path = here("workflow",
                                        "results",
                                        "anno_vcf_tbls"),
                            pattern = "HG002.*_anno.tsv") %>%
  set_names( ~ str_remove(., "_anno.tsv")) %>%
  map( ~ here("workflow", "results", "anno_vcf_tbls", .))


var_df <- var_tbl_files %>% 
  map_dfr(read_anno_tsv, .id = "benchmarkset") %>% 
  separate(benchmarkset, 
           into = c("giab_id", "ref", "benchmark", "bench_type"),
           sep = "_",
           remove = TRUE)
```

```{r}
glimpse(var_df)
```

### Writing To File 
```{r}
var_file <-  here("data","tidy","var_tbl.tsv.gz")

write_tsv(var_df, var_file)
```


## Strat Overlap Coverage

```{r}
strat_overlap_df <-
  list.files(
    path = here("workflow", "results", "strat_cov_tbls"),
    pattern = "_mrg_gene-plus-flank_cov.tsv",
    full.names = TRUE
  ) %>%
  {
    set_names(., str_extract(., "(?<=strat_cov_tbls/).*(?=_cov.tsv)"))
  } %>%
  map_dfr(
    read_tsv,
    col_types = "ciiiiid",
    col_names = c(
      "chrom",
      "start",
      "end",
      "n_regions",
      "bp_cov",
      "region_size",
      "overlap"
    ),
    .id = "id"
  ) %>%
  separate(
    col = id,
    into = c("strat", "ref", "mrg", "region"),
    sep = "_"
  ) %>%
  select(-mrg)
```


```{r}
glimpse(strat_overlap_df)
```


### Writing Strat Overlap Table to file 
```{r}
write_tsv(strat_overlap_df,
          here("data","tidy","strat_overlap.tsv.gz"))
```

## Assembly Coverage
```{r}
asm_cov_df <-
  list.files(
    path = here("workflow", "results", "asm_cov_tbls"),
    pattern = ".*_cov.tsv",
    full.names = TRUE
  ) %>%
  {
    set_names(., str_extract(., "(?<=asm_cov_tbls/).*(?=_cov.tsv)"))
  } %>%
  map_dfr(
    read_tsv,
    col_types = "ciiiiid",
    col_names = c(
      "chrom",
      "start",
      "end",
      "n_contigs",
      "bp_cov",
      "region_size",
      "overlap"
    ),
    .id = "id"
  ) %>%
  separate(
    col = id,
    into = c("giab_id", "ref", "asm","mrg", "region"),
    sep = "_"
  ) %>%
  select(-asm, -mrg)

asm_gene_cov_df <- gene_coords_df %>% 
  mutate(chrom = str_remove(coords, ":.*")) %>% 
  bind_cols(asm_cov_df)
```

__Data Check__: Making sure rows correctly bind
ASM coverage and gene coordinates should have the same 
```{r}
nrow(asm_cov_df)== nrow(gene_coords_df)
```
Corresponding columns should agree
```{r}
all(asm_gene_cov_df$ref...2 == asm_gene_cov_df$ref...8)
```

```{r}
all(asm_gene_cov_df$chrom...6 == asm_gene_cov_df$chrom...10)
```

```{r}
all(asm_gene_cov_df$start_flank == asm_gene_cov_df$start)
```

```{r}
all(asm_gene_cov_df$end_flank == asm_gene_cov_df$end)
```

Removing redundant columns
```{r}
asm_gene_cov_df <- asm_gene_cov_df %>% 
  rename(ref = ref...2 ) %>% 
  select(giab_id, gene, ref, start, end, 
         region, n_contigs, bp_cov, region_size, overlap)
```


### Table Structure
```{r}
glimpse(asm_gene_cov_df)
```


### Writing Assembly Coverage Table to file 
```{r}
write_tsv(asm_gene_cov_df,
          here("data","tidy","asm_coverage.tsv.gz"))
```

## Sanger Confirmation
Summary table at https://docs.google.com/spreadsheets/d/1_29bJkQ1snrKb9TGeCWh2hRl6Rx7tPVd/edit#gid=985369016


## Evaluation / Manual Curation
_Note from Slack_ RE: spreadsheet source 
    @nolson for the evaluation results figure, I think you'll want to use my curation results in columns T-U, unless they are blank, in which case you should use columns Q-S, in the GRCh37andGRCh38 tab in https://docs.google.com/spreadsheets/d/1Pn7WP78JfWKCO2Df31n_4gzDOwtP69flgDyyjeBS6JE/edit?usp=sharing.  This'll be a bit more complicated than previous figs, since we have the CommonFP/FN/FP_FN rows, so happy to chat about this when you get to it (edited) 
    

    
### Loading and Tidying
```{r}
eval_sheet <- "combined curation responses from benchmarking with sm variant v0.02.03.xlsx"
raw_bench_eval_df <- read_excel(here("data", "benchmark_evaluation", eval_sheet), 
                            sheet = "GRCh37andGRCh38", 
                            .name_repair = "universal") 
```
```{r}
glimpse(raw_bench_eval_df)
```

```{r}
tidy_bench_eval_df <- raw_bench_eval_df %>%
  rename(
    jz_benchmark_correct = "JZ..Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.benchmark...yes..no..or.unsure.",
    jz_query_correct = "JZ...Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.query.callset...yes..no..or.unsure.",
    benchmark_correct = "Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.benchmark...yes..no..or.unsure.",
    query_correct = "Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.query.callset...yes..no..or.unsure.",
    fp_or_fn = "Is.this.a.putative.FP..FN.or.genotype.error..FP_FN..in.your.calls.",
    truth_gt = "truth_gt.or.for..common..truth_gt.when.comparing.to.Roche.callset",
    query_gt = "query_gt.or.for..common..query_gt.when.comparing.to.Roche.callset",
    ref_base = ref,
    alt_base = alt
  ) %>%
  select(
    callset.ID,
    reference,
    SNP.or.INDEL,
    chrwithprefix,
    GRCh37_POS...8,
    GRCh38_POS,
    fp_or_fn,
    truth_gt,
    query_gt,
    ref_base,
    alt_base,
    jz_benchmark_correct,
    jz_query_correct,
    benchmark_correct,
    query_correct
  ) %>%
  ## Replacing benchmark_correct and query_correct with JZ responses
  mutate(
    benchmark_correct = if_else(
      is.na(jz_benchmark_correct),
      benchmark_correct,
      jz_benchmark_correct
    ),
    query_correct = if_else(is.na(jz_query_correct), query_correct, jz_query_correct)
  ) %>%
  select(-jz_benchmark_correct,-jz_query_correct) %>% 
  rename(ref = reference,
         callset = callset.ID,
         var_type = SNP.or.INDEL,
         GRCh37_POS = GRCh37_POS...8) %>% 
  mutate(benchmark_correct = tolower(benchmark_correct),
         query_correct = tolower(query_correct))
```

### Writing to file
```{r}
write_tsv(tidy_bench_eval_df, 
          here("data","tidy","manual_curation.tsv"))
```



# Getting MD5s
```{r}
list.files(here("data","tidy")) %>% 
  {tools::md5sum(here("data","tidy", .))}
```


# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```