---
title: "Making Tidy Dataset for MRG Manuscript"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(readxl)
library(here)
```

__TODOs__
-[] Add data descriptors

# Objective
Generate tidy data sets for generating manuscript figures and tables.

## Datasets

1. Medical Gene List (`data/tidy/mrg_list_info.tsv`): gene coordinates, gene list membership, and testing modality  
1. Genes coverage (`data/tidy/gene_asm_bench_coverage.tsv`): benchmark set and asm coverage information  
1. Gene stratification overlap (`data/tidy/gene_strat_overlap.tsv`): overlap fraction of difficult stratifications with medically relevant genes.


# Making Tidy Data Tables
Loading summary stat table generated by JW
```{r}
tbl_list <- list(
  GRCh37 = here("data", "gene_stat_tbls", "MRG_stats_table_GRCh37.tsv"),
  GRCh38 = here("data", "gene_stat_tbls", "MRG_stats_table_GRCh38.tsv")
)

c_types <- str_c("cddddc",
                 str_c(rep("d", 36), collapse = ""),
                 "ccc",
                 str_c(rep("d", 16), collapse = ""))

GRCh37_tbl <- read_tsv(tbl_list$GRCh37, col_types = c_types) %>%
  mutate(chrom = as.character(chrom)) %>% 
  rename(bp_hifiasm_dip.bed = "bp_coordinates_plus_segdups_plus_flanking_hifiasm_dip.bed",
         fraction_hifiasm_dip.bed = "fraction_coordinates_plus_segdups_plus_flanking_hifiasm_dip.bed")



c_types <- str_c("ccccddddc", str_c(rep("d", 51), collapse = ""))
GRCh38_tbl <- read_tsv(tbl_list$GRCh38, col_types = c_types)



mrg_stats_tbl <- bind_rows(GRCh37 = GRCh37_tbl, 
                           GRCh38 = GRCh38_tbl, 
                           .id = "ref")
```

```{r}
glimpse(mrg_stats_tbl)
```



## Medical Gene List
Extracting gene source information from main MRG table
```{r}
mrg_source_tbl <- mrg_stats_tbl %>% 
    select(gene, ref, chrom, start, end, 
           mandelker, cosmic, high_priority)
```


_DATA CHECK:_ Check for genes with multiple entries
33 genes with duplicate coordinate entries Using gene coordinates to differentiate.
__Follow-up__ Duplicate genes most likely due to multiple ENSAMBL gene id for the same gene name
```{r}
mrg_source_tbl %>%
  group_by(gene, ref) %>%
  mutate(count = n()) %>%
  filter(count != 1) %>%
  arrange(gene)
```

_DATA CHECK:_ Verifying only one entry per gene + coordinate combination. Should be 0 rows
```{r}
mrg_source_tbl %>% 
    count(gene, ref, chrom, start, end) %>% 
    filter(n > 1)
```

__QUESTION:__ Is ClinGen a testing modality? - Ask Steve During call 4/28 (Modalities should be targeted panels)

Information on testing modality and high priority gene list from Lincoln https://drive.google.com/drive/u/1/folders/1gUjMx24CkZCbt4vC46W3nQY_3DcjP4X9 


Gene testing modality from individual list generated by Justin Wagner in collaboration with Steve Lincoln.
```{r}
testing_modality_df <- list(ACMG = "ACMG_SF_2.0.tsv",
                             ACOG = "ACOG.tsv",
                             HTD = "Assorted_HTD.tsv",
                             ClinGen = "ClinGen_moderate_definitive_strong.tsv",
                             Counsyl = "Counsyl.tsv",
                             CPIC = "CPIC.tsv",
                             NCCN = "NCCN_ESMO.tsv") %>% 
    map(~here("data","mrg_lists","testing_modalities", .)) %>% 
    map_df(read_tsv, col_types = "c", .id = "testing_modality") %>% 
    count(Gene, testing_modality) %>% 
    pivot_wider(names_from = testing_modality,
                values_from = n,
                values_fill = 0) %>% 
    rename(gene = "Gene")
```

```{r}
mrg_info_tbl <- mrg_source_tbl %>%
    full_join(testing_modality_df)
```

_DATA CHECK:_ Checking that all genes in testing modality list are in mrg source list
All genes in the testing modality list are in the combined GRCh37 and GRCh38 gene lists. 

```{r}
nrow(mrg_info_tbl) == nrow(mrg_source_tbl)
```

Differing numbers of genes with and without testing modalities for the two reference genomes. 
This might be expected. 
Related to question above about which ref genome the testing modalities are relevant to.
```{r}
mrg_info_tbl %>% 
  mutate(test_modality_info = !is.na(ClinGen)) %>% 
  count(ref, test_modality_info)
```


Tidying tibble up for export
```{r}
tidy_mrg_info_tbl <- mrg_info_tbl %>%
    ## replacing NAs from testing modality with 0s
    mutate(across(where(is.integer), replace_na, 0))
```

### Example Gene
```{r}
tidy_mrg_info_tbl %>%
  filter(gene == "OR4F5") %>%
  DT::datatable()
```

### Writing To File 
```{r}
write_tsv(tidy_mrg_info_tbl,
          here("data","tidy","gene_list_info.tsv"))
```


## Benchmark and Assembly Coverage
Generating tidy table with gene and exon coverage for the medically relevant genes benchmark, v4 small variant benchmark, and hifi assembly.

### Loading MRG Benchmark Gene Lists
```{r}
## Gene List Files
mrg_gene_beds <- list(
  GRCh37 = here(
    "data",
    "mrg_benchmarkset",
    "HG002",
    "GRCh37",
    "intermediate_files",
    "HG002_GRCh37_MRG_no_MHC.bed"
  ),
  GRCh38 = here(
    "data",
    "mrg_benchmarkset",
    "HG002",
    "GRCh38",
    "intermediate_files",
    "HG002_GRCh38_MRG_no_MHC.bed"
  )
)
  


mrg_genes_df <- mrg_gene_beds %>%
  map_dfr(
    read_tsv,
    col_names = c("chrom", "start", "end", "gene"),
    col_types = "cddc",
    .id = "ref"
  ) %>% 
  mutate(in_mrg = TRUE)
```




__QUESTION__ Are these 440 NA values expected for GRCh38 - Out dated
Number of entries with NA values for MRG v0.03 and SV 0.02
```{r}
mrg_stats_tbl %>% 
  group_by(ref) %>% 
  summarise(across(where(is.numeric), ~ sum(is.na(.x)))) %>% 
  pivot_longer(cols = -ref) %>% 
  group_by(name) %>% 
  mutate(total = sum(value)) %>% 
  filter(total != 0) %>% 
  arrange(-total, name) %>% 
  select(-total)
```

```{r}
na_genes <- mrg_stats_tbl %>% 
  filter(is.na(fraction_exon_covered_MRG_small_variant_v0.03_benchmark)) %>% 
  {.$gene} %>% unique()

mrg_stats_tbl %>% 
  filter(gene %in% na_genes,
         str_detect(chrom, "[MTXY]", negate = TRUE)) %>% 
  arrange(gene)
```


```{r}
mrg_bench_asm_df <- mrg_stats_tbl %>% 
  left_join(mrg_genes_df) %>% 
  mutate(in_mrg = if_else(is.na(in_mrg), FALSE, in_mrg)) %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end)) %>% 
    select(gene, ref, coords, in_mrg,
           contains("V4"), 
           contains("MRG"), 
           -starts_with("bp_"), 
           -contains("CNV"), 
           -contains("stratification"),
           fraction_hifiasm_dip.bed,
          ## excluding small MRG v0.2 var counts and coverage
          -fraction_MRG_benchmark_bed,
          -fraction_exon_covered_MRG_benchmark,
           -num_MRG_small_variant_v0.2_variants_in_gene,
           -num_MRG_small_variant_v0.2_variants_in_exons,
          ## excluding SVs MRG v0.1 counts
          -num_MRG_SV_v0.1_in_gene,
          -num_MRG_SV_v0.1_benchmark_variants_in_exons) %>% 
  ## Renaming columns for consistent formatting
    rename(v4_gene_cov = fraction_HG002_v4.2.1,
           v4_exon_cov = fraction_exon_covered_v4.2.1,
           v4_gene_vars = num_v4.2.1_small_variants_in_gene,
           v4_exon_vars = num_v4.2.1_variants_in_exons,
           mrg_gene_cov = fraction_MRG_small_variant_v0.03,
           mrg_exon_cov = fraction_exon_covered_MRG_small_variant_v0.03_benchmark,
           mrg_gene_vars = num_MRG_small_variant_v0.03_variants_in_gene,
           mrg_exon_vars = num_MRG_small_variant_v0.03_variants_in_exons,
           mrg_gene_svs = num_MRG_SV_v0.02_SVs_in_gene,
           mrg_exon_svs = num_MRG_SV_v0.02_variants_in_exons,
           hifi_gene_cov = fraction_hifiasm_dip.bed) %>% 
    pivot_longer(cols = -c("gene", "ref", "coords", "in_mrg"),
                 names_to = c("benchmark", "region","metric"),
                 names_sep = "_", values_to = "value") %>%
    pivot_wider(id_cols = c(gene, ref, coords, in_mrg, benchmark, region),
                names_from = metric, values_from = value)

mrg_bench_asm_df
```


### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_bench_asm_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable(rownames = FALSE)
```

### Writing To File 
```{r}
bench_cov_file <-  here("data","tidy","gene_asm_bench_coverage.tsv")

write_tsv(mrg_bench_asm_df, bench_cov_file)
```

## MRG Benchmark Coverage
```{r}
## Gene Coverage
gene_cov_files <- list.files(path = here("data", "gene_stat_tbls", "tbl_inputs"),
                            pattern = "mrg.*gene_cov.tsv") %>% 
  set_names(~str_remove(., "_cov.tsv")) %>% 
  map(~here("data", "gene_stat_tbls", "tbl_inputs", .))

mrg_gene_cov_df <- gene_cov_files %>% 
  map_dfr(read_tsv, 
          col_names = c("chrom","start","end","gene",
                        "n_regions", "bp_cov", "gene_size", "cov"),
          col_types = "ciiciiid",
          .id = "covset") %>% 
    separate(col = covset, 
           into = c("giab_id", "ref","benchmark","bench_type","region"),
           remove = TRUE) %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end))

## Gene Coordinates
mrg_gene_coords_df <- mrg_gene_cov_df %>% 
  select(gene, ref, chrom, coords) %>% 
  distinct()
```


```{r}
## Exon Coverage
exon_cov_files <- list.files(path = here("data", "gene_stat_tbls", "tbl_inputs"),
                            pattern = "mrg.*exon_cov.tsv") %>% 
  set_names(~str_remove(., "_cov.tsv")) %>% 
  map(~here("data", "gene_stat_tbls", "tbl_inputs", .))

mrg_exon_cov_df <- exon_cov_files %>% 
  map_dfr(read_tsv, 
          ## ensembl colnames are assumed
          col_names = c("chrom","start","end","gene", 
                        "ensembl_gene_id", "ensembl_exon_id",
                        "n_regions", "bp_cov", "exon_size", "frac_cov"),
          col_types = "ciiccciiid",
          .id = "covset") %>% 
  group_by(covset, chrom, gene, ensembl_gene_id) %>%
  summarise(total_bp_cov = sum(bp_cov),
            total_exon_size = sum(exon_size)) %>%
  mutate(cov = total_bp_cov/ total_exon_size) %>%
  separate(col = covset,
           into = c("giab_id", "ref","benchmark","bench_type","region"),
           remove = TRUE) %>% 
    left_join(mrg_gene_coords_df)
```
```{r}
mrg_gene_cov_df %>% filter(gene == "ALG9")
```



__QUESTION__ Check why duplicate genes here - is it the same, should be addressed after fixing duplicate gene names in Gene bed
```{r}
nrow(mrg_exon_cov_df)
```

```{r}
nrow(mrg_gene_cov_df)
```
```{r}
mrg_gene_cov_df
```

```{r}
## Combined Cov DF making data frames consistent with DF defined below
# Only keeping genes in MRG benchmark set 
mrg_cov_df <- bind_rows(mrg_exon_cov_df, mrg_gene_cov_df) %>%
  ungroup() %>%
  select(gene, ref, coords, benchmark, bench_type, region, cov) %>%
  ## Annotating with in MRG info
    left_join(mrg_genes_df) %>%
  filter(in_mrg) %>%
  select(-chrom, -start, -end, -in_mrg)
```

```{r}
glimpse(mrg_cov_df)
```


__QUESTION:__ Genes with duplicate entries for gene coverage bed - addressing with input gene list bed fix
```{r}
dup_genes_df <- bind_rows(mrg_exon_cov_df, mrg_gene_cov_df) %>% 
  select(gene, ref, chrom, start, end, benchmark, bench_type, region, cov) %>% 
  arrange(gene) %>% 
  count(gene, ref, chrom, bench_type, region) %>% 
  filter(n != 1)

dup_genes <- unique(dup_genes_df$gene)

dup_genes_df
```
__TODO__ Fix bug - Only 4 genes in MRG benchmark affected by bug - addressing with input gene list bed fix
```{r}
bind_rows(mrg_exon_cov_df, mrg_gene_cov_df) %>% 
  filter(gene %in% dup_genes, 
         cov > 0) %>% 
  arrange(gene)
```


### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_cov_df %>% 
    filter(gene == "AGL") %>% 
    DT::datatable(rownames = FALSE)
```

### Writing To File 
```{r}
mrg_cov_file <-  here("data","tidy","mrg_bench_coverage.tsv")

write_tsv(mrg_cov_df, mrg_cov_file)
```

## MRG Variants Tables
__Scratch__
- Check with JZ about variant lengths when multiple alternate alleles

```{r}
mrg_grch38_smvar_tbl <- read_tsv(here("scratch","anno_vcfs","grch38_smvar.tbl"),
                                 col_names = c("CHROM","POS","GT","TYPE","GENE","EXON","REF","ALT"),
                                 col_types = "cicccccc")

mrg_grch38_sv_tbl <- read_tsv(here("scratch", "anno_vcfs","grch38_sv.tbl"),
                              col_names = c("CHROM","POS","GT","TYPE","REPTYPE","BREAKSIMLENGTH","GENE","EXON","REF","ALT"),
                                 col_types = "cicccicccc")

## Calculating INDEL Variant Size
get_var_size <- function(REF, ALT){
  ref_bp <- nchar(REF)
  
  ## Dealing with multiple ALTs
  # --- using largest SV
  if(str_detect(ALT, ",")){
    alts <- unlist(str_split(ALT, ","))
    alts_bp <- nchar(alts)
    var_sizes <- alts_bp - ref_bp
    
    var_size <- var_sizes[abs(var_sizes) == max(abs(var_sizes))]
    
    ## returning first entry value if the two genotypes are the same size
    return(var_size[1])
  }
  return(nchar(ALT) - ref_bp)
}


smvar_df <- mrg_grch38_smvar_tbl %>% 
  rowwise() %>% 
  mutate(var_size = get_var_size(REF,ALT)) %>% 
  filter(abs(var_size) < 50) %>% 
  select(-REF, -ALT)

sv_df <- mrg_grch38_sv_tbl %>% 
  rowwise() %>% 
  mutate(var_size = get_var_size(REF,ALT)) %>% 
  select(-REF, -ALT)

mrg_var_df <- bind_rows(smvar = smvar_df, sv = sv_df, .id = "bench_type")
```

```{r}
glimpse(mrg_var_df)
```

### Writing To File 
```{r}
mrg_var_file <-  here("data","tidy","mrg_var_tbl.tsv")

write_tsv(mrg_var_df, mrg_var_file)
```

## Stratification Coverage
```{r}
colnames(mrg_stats_tbl)
```

```{r}
mrg_strat_df <- mrg_stats_tbl %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end)) %>%  
    select(ref, gene, coords, 
           contains("fraction"),
           -contains("V4.2"), 
           -contains("MRG"), 
           -fraction_hifiasm_dip.bed) %>% 
    rename(giabSV = fraction_in_Tier1andTier2_SV_0.6_slop150_stratification,
           repeatsGt10kb = fraction_in_AllRepeats_gt_10kb,
           asmCNV = fraction_in_assembly_cnv_stratification,
           vdj = fraction_in_vdj_stratification,
           inversion = fraction_in_inversions_stratification,
           segdupHiCntHiPct = fraction_in_segmental_duplications_gt5_percent_identity_gte_990_stratification,
           segdup = fraction_in_segmental_duplications_stratification,
           lowmapAndSegDup = fraction_in_low_mappability_and_segmental_duplications_stratification,
           intersectCNV =fraction_in_illumina_ccs_ont_cnv_stratification,
           outlierCNV = fraction_in_ccs_ont_elliptical_outlier_cnv_stratification,
           allDiff = fraction_in_HG002_GIABv4.1_complexandSVs_alldifficultregions_stratification,
           refGaps = fraction_in_reference_gaps_stratification,
           CNVandSVs = fraction_in_HG002_GIABv4.1_CNVsandSVs_stratification) %>% 
    pivot_longer(cols = -c("ref", "gene", "coords"), 
                 names_to = "strat", values_to = "overlap")
```

```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_strat_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable()
```

### Writing Gene List Table to file 
```{r}
write_tsv(mrg_strat_df,
          here("data","tidy","gene_strat_overlap.tsv"))
```


## Sanger Confirmation
Summary table at https://docs.google.com/spreadsheets/d/1_29bJkQ1snrKb9TGeCWh2hRl6Rx7tPVd/edit#gid=985369016


## Evaluation / Manual Curation
_Note from Slack_ RE: spreadsheet source 
    @nolson for the evaluation results figure, I think you'll want to use my curation results in columns T-U, unless they are blank, in which case you should use columns Q-S, in the GRCh37andGRCh38 tab in https://docs.google.com/spreadsheets/d/1Pn7WP78JfWKCO2Df31n_4gzDOwtP69flgDyyjeBS6JE/edit?usp=sharing.  This'll be a bit more complicated than previous figs, since we have the CommonFP/FN/FP_FN rows, so happy to chat about this when you get to it (edited) 
    

    
### Loading and Tidying
```{r}
eval_sheet <- "combined curation responses from benchmarking with sm variant v0.02.03.xlsx"
raw_bench_eval_df <- read_excel(here("data", "benchmark_evaluation", eval_sheet), 
                            sheet = "GRCh37andGRCh38", 
                            .name_repair = "universal") 
```
```{r}
glimpse(raw_bench_eval_df)
```

```{r}
tidy_bench_eval_df <- raw_bench_eval_df %>%
  rename(
    jz_benchmark_correct = "JZ..Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.benchmark...yes..no..or.unsure.",
    jz_query_correct = "JZ...Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.query.callset...yes..no..or.unsure.",
    benchmark_correct = "Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.benchmark...yes..no..or.unsure.",
    query_correct = "Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.query.callset...yes..no..or.unsure.",
    fp_or_fn = "Is.this.a.putative.FP..FN.or.genotype.error..FP_FN..in.your.calls.",
    truth_gt = "truth_gt.or.for..common..truth_gt.when.comparing.to.Roche.callset",
    query_gt = "query_gt.or.for..common..query_gt.when.comparing.to.Roche.callset",
    ref_base = ref,
    alt_base = alt
  ) %>%
  select(
    callset.ID,
    reference,
    SNP.or.INDEL,
    chrwithprefix,
    GRCh37_POS...8,
    GRCh38_POS,
    fp_or_fn,
    truth_gt,
    query_gt,
    ref_base,
    alt_base,
    jz_benchmark_correct,
    jz_query_correct,
    benchmark_correct,
    query_correct
  ) %>%
  ## Replacing benchmark_correct and query_correct with JZ responses
  mutate(
    benchmark_correct = if_else(
      is.na(jz_benchmark_correct),
      benchmark_correct,
      jz_benchmark_correct
    ),
    query_correct = if_else(is.na(jz_query_correct), query_correct, jz_query_correct)
  ) %>%
  select(-jz_benchmark_correct,-jz_query_correct) %>% 
  rename(ref = reference,
         callset = callset.ID,
         var_type = SNP.or.INDEL,
         GRCh37_POS = GRCh37_POS...8) %>% 
  mutate(benchmark_correct = tolower(benchmark_correct),
         query_correct = tolower(query_correct))
```

### Writing to file
```{r}
write_tsv(tidy_bench_eval_df, 
          here("data","tidy","manual_curation.tsv"))
```



# Getting MD5s
```{r}
list.files(here("data","tidy")) %>% 
  {tools::md5sum(here("data","tidy", .))}
```


# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```