---
title: "Medical Gene Benchmark Inclusion and Overlap"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
library(readxl)
library(ggpubr)
```

## Loading and Tidying
# Loading and Tidying Data
```{r}
## Gene source and testing modality information
mrg_info_tbl <- here("data","tidy","gene_list_info.tsv") %>% 
    read_tsv(col_types = "cccddcccddddddd")

## Gene Stratfication Overlap
gene_strat_df <- read_tsv(here("data","tidy","gene_strat_overlap.tsv"))

## TODO - replace with new updated coverage data
## Benchmark and Asm Coverage information 
bench_asm_cov_df <- here("data","tidy","gene_asm_bench_coverage.tsv") %>% 
    read_tsv(col_types = "ccclccddd")

## Benchmark Variant tbl
mrg_var_df <- read_tsv(here("data","tidy","mrg_var_tbl.tsv"),
                       col_types = "cciccccici")
```

Combining Benchmark Coverage and Strat data frames
```{r}
mrg_gene_cats <- bench_asm_cov_df %>% 
    filter(str_detect(coords, "[MTXY]",negate = TRUE),
           ref == "GRCh38",
           region == "gene") %>% 
    select(-vars, -svs) %>% 
    pivot_wider(names_from = benchmark, values_from = cov) %>% 
    mutate(gene_cat = case_when(in_mrg ~ "MRG",
                                v4 > 0.9 ~ "V4",
                                v4 < 0.9 & hifi < 1 ~ "Not Benchmarked",
                                mrg == 0 & hifi == 1 ~ "Excluded?",
                                TRUE ~ "???"))

bench_cov_anno_df <- mrg_gene_cats %>% 
    left_join(gene_strat_df)
```

## Panels

### V4 Coverage


```{r}
v4_cov_df <- bench_asm_cov_df %>% 
    ## Excluding mito and sex chromosomes
    filter(ref == "GRCh38",
           str_detect(coords, "[MTXY]",negate = TRUE), 
           region == "gene") %>% 
     pivot_wider(id_cols = c("gene","ref","coords", "in_mrg"), 
                 names_from = benchmark, values_from = cov)

n_cov_df <- v4_cov_df %>% 
    mutate(cov_lvl = if_else(v4 > 0.9, "high v4 cov", "low v4 cov")) %>% 
    count(ref, cov_lvl)

v4_low_cov_df <- bench_asm_cov_df %>% 
    ## Excluding mito and sex chromosomes
    filter(ref == "GRCh38",
           str_detect(coords, "[MTXY]",negate = TRUE), 
           region == "gene") %>% 
     pivot_wider(id_cols = c("gene","ref","coords","in_mrg"), 
                 names_from = benchmark, values_from = cov) %>% 
    filter(v4 < 0.9) %>% 
    mutate(benchmarked = if_else(in_mrg, "in MRG","not in MRG"),
           benchmarked = factor(benchmarked, levels = c("not in MRG", "in MRG")))
```

```{r}
n_high_v4_cov <- n_cov_df$n[n_cov_df$cov_lvl == "high v4 cov"]
n_low_v4_cov <- n_cov_df$n[n_cov_df$cov_lvl == "low v4 cov"] 

(
    v4_cov_plt <- ggplot(v4_low_cov_df) +
        geom_histogram(
            aes(x = v4, fill = benchmarked),
            color = "grey20",
            size = 0.25,
            bins = 90
        ) +
        geom_text(aes(
            x = 0.05,
            y = 42,
            label = str_c(n_high_v4_cov,
                          " High Coverage Genes (not shown)"),
            hjust = 0
        )) +
        geom_text(aes(
            x = 0.05,
            y = 38,
            label = str_c(n_low_v4_cov, 
                          " Low Coverage Genes"),
            hjust = 0
        )) +
        theme_bw() +
        labs(x = "Gene Coverage by V4 Benchmark", y = "Number of Genes") +
        theme(legend.position = c(0.25, 0.65),
              legend.title = element_blank())
)
```
__DEBUGME__ Fix issue with number of genes in MRG
```{r}
v4_low_cov_df
```
Bug with processing `bench_asm_cov_df`
```{r}
bench_asm_cov_df %>% 
    select(gene, ref, in_mrg) %>% 
    distinct() %>% 
    count(ref, in_mrg)
```


### MRG Gene Coverage

```{r}
mrg_gene_cov <- bench_cov_anno_df %>% 
    filter(gene_cat == "MRG") %>% 
    select(gene, ref, coords) %>% 
    distinct() %>% 
    left_join(bench_asm_cov_df) %>% 
    filter(ref == "GRCh38", 
           str_detect(coords, "[MTXY]",negate = TRUE),
           benchmark == "mrg")

(mrg_gene_cov_plt <- ggplot(mrg_gene_cov) + 
    geom_histogram(aes(x = cov, fill = region), 
                   position = "dodge",
                   color = "grey40",
                   size = 0.25, bins = 50) + 
    theme_bw() +
    labs(x = "Benchmark Coverage", y = "Number of Genes", fill = "") +
    theme(legend.position = c(0.15,0.9), 
          legend.direction = "horizontal", legend.background = element_blank()))
```

### MRG v. V4 Coverage (TO REPLACE V4 and MRG Coverage)

### Difficult Gene Overlap
```{r}
(diff_overlap_plt <- bench_cov_anno_df %>% 
    filter(ref == "GRCh38",
           gene_cat %in% c("V4", "MRG","Not Benchmarked"),
        str_detect(coords, "[MTXY]",negate = TRUE),
        strat == "allDiff") %>% 
    mutate(gene_cat = factor(gene_cat, 
                             level = c("V4", "MRG","Not Benchmarked"),
                             labels = c(">90% coverage by V4.2.1",
                                        "in MRG Benchmark",
                                        "Not Benchmarked"))
           ) %>% 
    ggplot() + 
    geom_histogram(aes(x = overlap, fill = gene_cat), 
                   color = "grey40", size = 0.25, bins = 100) + 
    facet_wrap(~gene_cat, ncol = 1, scales = "free_y") + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "Fraction of gene with challenging sequence or variants", 
         y = "Number of Genes",
         fill = "Benchmark"))
```

### Small Variant Size Distribution
```{r}
(smv_size_dist_plt <- mrg_var_df %>% 
    filter(TYPE == "INDEL", bench_type == "smvar") %>% 
    mutate(gene_part = if_else(EXON == ".", "intron","exon"),
           gene_part = factor(gene_part, levels = c("intron","exon"))) %>% 
    ggplot() + 
    geom_bar(aes(x = var_size, fill = gene_part)) + 
        scale_fill_brewer(type = "qual", palette = 3) +
    theme_bw() +
    labs(x = "Small INDEL Size (bp)", y = "# of Variants", fill = "Gene Part") + 
    theme(legend.position = c(0.15, 0.8), legend.background = element_blank()))
```

### Large Variant Size Distribution 


```{r}
breaks <- c(50, 100, 500, 1000, 5000, 10000)
log_breaks <- log10(breaks)

sv_log_breaks <- sort(c(-log_breaks + log10(25), 0,  log_breaks - log10(25)))

sv_breaks <- sort(c(-breaks, 0, breaks))

(sv_size_dist_plt <- mrg_var_df %>% 
    filter(TYPE == "INDEL", bench_type == "sv") %>% 
    mutate(gene_part = if_else(EXON == ".", "intron","exon"),
           gene_part = factor(gene_part, levels = c("intron","exon"))) %>%         
    ## Log scale var length with 1 offset 
    mutate(var_size = if_else(var_size < 0, 
                              -log10(abs(var_size)) + log10(25),
                              log10(var_size) - log10(25))) %>% 
    ggplot() + 
    geom_histogram(aes(x = var_size, fill = gene_part), 
                   color = "grey40", size = 0.25, binwidth = 0.125) +
    scale_x_continuous(breaks = sv_log_breaks,
                       labels = sv_breaks) + 
        scale_fill_brewer(type = "qual", palette = 3) + 
    theme_bw() +
    theme(axis.text.x = element_text(angle = -45, hjust = 0),
          legend.position = c(0.15, 0.8), legend.background = element_blank()) + 
    labs(x = "Large SV Size (bp)", y = "# of Variants"))
```


# Combined Figure
```{r}
(combined_fig <-
     ggarrange(
         ggarrange(
             ggarrange(
                 v4_cov_plt,
                 mrg_gene_cov_plt,
                 ncol = 1,
                 labels = "AUTO"
             ),
             diff_overlap_plt,
             labels = c("", "C"),
             nrow = 1
         ),
         ggarrange(
             smv_size_dist_plt,
             sv_size_dist_plt + rremove(object = "legend"),
             nrow = 1,
             labels = c("D", "E")
         ),
         nrow = 2,
         heights = c(3,2)
     )
)
```

```{r}
ggsave(filename = here("figures", "fig2_mrg_inclusion_overlap.png"),
       plot = combined_fig, width = 8, height = 10, dpi = 300)
```




# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```