---
title: "MRG Benchmark Set Summary Statistics"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

__Next Steps__
- Add tidying steps for all diff coverage
- Add V0.6 SV to snakemake pipeline for coverage calculations
- Add V4 and V0.6 to var stats table

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
```

# Objective

Metrics to include - break down relevant metrics by intron/ exon and for high priority genes
- Total bases in MRG and high priority
    - Calculate total bases for can use bench coverage table, after adding total bases and annotating high priority genes
        - MRG regions
        - MRG overlap high priority 
        - V4 overlap med genes
        - v4 overlap high priority
- Variant Counts
    - # of SNVs, # of small indels, # of large indels - bcftools
    - Calculate using bcftools stats
        - MRG smallvar
        - MRG SV
        - MRG small var overlap with high priority 
        - MRG SV overlap with high priority
        - V4 overlap with MRG
        - V4 overlap with high priority
        - v0.6 sv overlap with MRG
        - v0.6 sv overlap with high priority
- # bp in low map and segdups
    - Calculate gene + lowmap & segdup coverage bed

# Loading Data
```{r}
## Gene List Info - for high priority genes
mrg_info_tbl <- here("data", "tidy", "gene_list_info.tsv") %>%
    read_tsv(col_types = "ccccciiiiiii")

## Benchmark Included Bases
bench_cov_df <- here("data","tidy","bench_coverage.tsv") %>% 
    read_tsv(col_types = "ccccccdil")

## Benchmark Difficult Bases
bench_diff_df <- here("data", "tidy", "bench_diff_bases.tsv") %>% 
    read_tsv(col_types = "ccccccdi")

## Variant Stats Table
var_df <- read_tsv(here("data","tidy","var_tbl.tsv"),
                       col_types = "ccccciccccici"
                   )
```


# Extracting Stats

## Total bases in MRG gene list for GIAB Benchmarks
```{r}
mr_gene_bases_df <- bench_cov_df %>% 
     group_by(ref, benchmark, bench_type, region) %>% 
     summarise(total_bp = sum(bp_cov)) %>% 
     # filter((benchmark == "v0.6" & ref == "GRCh37") | 
     #        (benchmark != "v0.6" & ref == "GRCh38")) %>% 
     pivot_wider(names_from = "region", values_from = "total_bp") %>% 
     arrange(-gene)

DT::datatable(mr_gene_bases_df)

```
- total bases in high priority gene list for mrg and v4 benchmarks

```{r}
hp_gene_bases_df <- bench_cov_df  %>%
    left_join(mrg_info_tbl) %>%
     # filter(high_priority == "yes") %>%
    group_by(ref, benchmark, bench_type, region, high_priority) %>%
    summarise(total_bp = sum(bp_cov)) %>%
    # filter(ref == "GRCh38") %>%
    pivot_wider(names_from = "region", values_from = "total_bp") %>%
    arrange(bench_type, benchmark, high_priority)

DT::datatable(hp_gene_bases_df)

```

- Bases in difficult regions
```{r}
mr_diff_bases_df <- bench_diff_df %>%
    mutate(region = str_extract(region, "(?<=-).*(?=-)")) %>%
    group_by(ref, benchmark, bench_type, region) %>%
    summarise(total_bp = sum(bp_cov)) %>%
    # filter(ref == "GRCh38") %>%
    pivot_wider(names_from = "region", values_from = "total_bp")

DT::datatable(mr_diff_bases_df)

```

- Variant Counts
```{r}
var_count_tbl <- var_df %>% 
    mutate(region = if_else(EXON == ".","exon","intron")) %>% 
    group_by(ref, benchmark, bench_type, TYPE, region) %>% 
    count(name = "n_vars") %>% 
    pivot_wider(names_from = "region", 
                values_from = "n_vars", 
                values_fill = 0) %>% 
    arrange(bench_type, TYPE, ref) %>% 
    mutate(gene = exon + intron)
DT::datatable(var_count_tbl)
```


# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```