---
title: "MRG Benchmark Set Summary Statistics"
author: "Nate Olson"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
```

# Objective

Metrics to include - break down relevant metrics by intron/ exon and for high priority genes  

- Total bases in MRG and high priority  
    - Calculate total bases for can use bench coverage table, after adding total bases and annotating high priority genes   
        - MRG regions  
        - MRG overlap high priority   
        - V4 overlap med genes  
        - v4 overlap high priority  

- Variant Counts # of SNVs, # of small indels, # of large indels - bcftools  
    - Calculate using bcftools stats  
        - MRG smallvar  
        - MRG SV  
        - MRG small var overlap with high priority  
        - MRG SV overlap with high priority  
        - V4 overlap with MRG  
        - V4 overlap with high priority  
        - v0.6 sv overlap with MRG  
        - v0.6 sv overlap with high priority  

- Number of bp in low map and segdups  
    - Calculate gene + lowmap & segdup coverage bed  

# Loading Data
```{r}
## Gene List Info - for high priority genes
mrg_info_tbl <- here("data", "tidy", "gene_list_info.tsv") %>%
    read_tsv(col_types = "ccccciiiiiii")

## Benchmark Included Bases
bench_cov_df <- here("data","tidy","bench_coverage.tsv") %>% 
    read_tsv(col_types = "ccccccdil")


## Benchmark Difficult Bases
bench_diff_df <- here("data", "tidy", "bench_diff_bases.tsv") %>% 
    read_tsv(col_types = "ccccccdi")

## Variant Stats Table
var_df <- read_tsv(here("data","tidy","var_tbl.tsv"),
                       col_types = "ccccciccccici"
                   )
```
Data frame with just genes in mrg benchmark
```{r}
mrg_bench_genes_df <- bench_cov_df %>% 
    select(gene, ref, in_mrg) %>% 
    distinct() %>% 
    filter(in_mrg == TRUE) %>% 
    select(-in_mrg)

```


# Extracting Stats

## Total bases in MRG gene list for GIAB Benchmarks
```{r}
mr_gene_bases_df <- bench_cov_df %>% 
     ## Only genes in mrg benchmark
     right_join(mrg_bench_genes_df) %>% 
     ## Total bases by benchmark and region
     group_by(ref, benchmark, bench_type, region) %>% 
     summarise(total_bp = sum(bp_cov)) %>% 
     pivot_wider(names_from = "region", values_from = "total_bp") %>%
    ungroup() %>%
     arrange(bench_type, benchmark, ref) %>%
    select(ref, benchmark, bench_type, gene, exon, intron)

DT::datatable(mr_gene_bases_df,
              caption = "Medically relevant gene bases included in GIAB benchmarks by gene region.")

```
## total bases in high priority gene list for mrg and v4 benchmarks

```{r}
hp_gene_bases_df <- bench_cov_df  %>%
         ## Only genes in mrg benchmark
     right_join(mrg_bench_genes_df) %>% 
    left_join(mrg_info_tbl) %>%
     filter(high_priority == "yes") %>%
    group_by(ref, benchmark, bench_type, region, high_priority) %>%
    summarise(total_bp = sum(bp_cov)) %>%
    pivot_wider(names_from = "region", values_from = "total_bp") %>%
    arrange(bench_type, benchmark)  %>% 
    select(ref, benchmark, bench_type, gene, exon, intron)

DT::datatable(hp_gene_bases_df, 
              caption = "High priority gene bases included in GIAB benchmarks by gene region")

```

## Bases in difficult regions

__Note__ Can add high priority bases
```{r}
mr_diff_bases_df <- bench_diff_df %>%
         ## Only genes in mrg benchmark
     right_join(mrg_bench_genes_df) %>% 
    mutate(region = str_extract(region, "(?<=-).*(?=-)")) %>%
    group_by(ref, benchmark, bench_type, region) %>%
    summarise(total_bp = sum(bp_cov)) %>%
    pivot_wider(names_from = "region", values_from = "total_bp")  %>%
    arrange(bench_type, benchmark)  %>% 
    select(ref, benchmark, bench_type, gene, exon, intron)

DT::datatable(mr_diff_bases_df,
             caption = "Medically relevant genes bases in difficult regions included in GIAB benchmarks by gene region")
```

```{r}
(v06_df <- var_df %>% 
    filter(!(bench_type == "SV" & abs(var_size) < 50)) %>% 
     filter(bench_type == "SV") %>% 
     arrange(abs(var_size))
)
    
```

## Variant Counts
```{r}

var_count_tbl <- var_df %>% 
    ## Limiting to genes in mrg benchmark
    rename(gene = GENE) %>% 
    right_join(mrg_bench_genes_df) %>% 
    ## Excluding variants in SV benchmark less thank 50 bp
    filter(!(bench_type == "SV" & abs(var_size) < 50)) %>% 
    mutate(region = if_else(EXON == ".","exon","intron")) %>% 
    group_by(ref, benchmark, bench_type, TYPE, region) %>% 
    count(name = "n_vars") %>% 
     ## Only genes in mrg benchmark
    pivot_wider(names_from = "region", 
                values_from = "n_vars", 
                values_fill = 0) %>% 
    arrange(bench_type, TYPE, ref) %>% 
    mutate(gene = exon + intron)  %>%
    arrange(bench_type, benchmark)  %>% 
    select(ref, benchmark, bench_type, gene, exon, intron)

DT::datatable(var_count_tbl,
              caption = "Variants in medically relevant genes included in GIAB benchmarks by gene region")
```
```{r}
sv_only_df <- var_df %>% 
    ## Limiting to genes in mrg benchmark
    rename(gene = GENE) %>% 
    right_join(mrg_bench_genes_df) %>% 
    ## Excluding variants in SV benchmark less thank 50 bp
    filter(bench_type == "SV" & abs(var_size) >= 50) 

sv_only_df %>% 
    mutate(region = if_else(EXON == ".","exon","intron")) %>% 
    group_by(ref, benchmark, bench_type, TYPE, region) %>% 
    count(name = "n_vars") %>% 
     ## Only genes in mrg benchmark
    pivot_wider(names_from = "region", 
                values_from = "n_vars", 
                values_fill = 0) %>% 
    arrange(bench_type, TYPE, ref) %>% 
    mutate(gene = exon + intron)  %>%
    arrange(bench_type, benchmark)  %>% 
    select(ref, benchmark, bench_type, gene, exon, intron)
```

# Summary Table
- GRCh38 Only
- Rows - benchmarks
- Columns - bases and variants or all mrg, high priority

# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```