---
title: "Medically Relevant Genes Characterizations"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
library(readxl)
```

## Medical Genes Benchmark

**TODO** Add plot with overlap distributions for 4 gene cats, lowmap as
shortcut **Panels**

{DRAFT} A. Gene assembly coverage - flanking diagram  - Illustrator diagram
{DRAFT} B. Medical composition - Upset plot with testing modalities 
{DRAFT} C. Small variant benchmark (poster figure) over GATK  
D. Sanger confirmation?   
E. Coverage by gene part (exon, introns,
flanking)    
F. small var sv and indel size distributions

**Key Points**\
- Identification of candidate genes\
- 274 genes with know significant relevance\
- Not all genes are benchmarkable - coverage of genes and number of
small and structural variants, bases / variants excluded due to
different filters.

# Loading and Tidying Data
```{r}
## Gene source and testing modality information
mrg_info_tbl <- here("data","tidy","gene_list_info.tsv") %>% 
    read_tsv(col_types = "cccddcccllddddddd")

## Gene Stratfication Overlap
gene_strat_df <- read_tsv(here("data","tidy","gene_strat_overlap.tsv"))

## Benchmark and Asm Coverage information
bench_asm_cov_df <- here("data","tidy","gene_asm_bench_coverage.tsv") %>% 
    read_tsv(col_types = "cccccddd")

## MRG versus GATK
# Extracting Column Names
gatk_colnames <- here("data","benchmark_evaluation",
                      "smvar HG002 MRG v0.03.00 evaluation.xlsx") %>% 
    read_excel(sheet = "summary stats NYGC", range = "B1:AR2", na = "NA")

## Getting benchmark results
gatk_bench_df <- here("data",
                      "benchmark_evaluation",
                      "smvar HG002 MRG v0.03.00 evaluation.xlsx") %>%
    read_excel(
        sheet = "summary stats NYGC",
        range = "B32:AR45",
        na = "NA",
        col_names = colnames(gatk_colnames)
    )
```

Combining Benchmark Coverage and Strat data frames
```{r}
mrg_gene_cats <- bench_asm_cov_df %>% 
    filter(str_detect(coords, "[MTXY]",negate = TRUE),
           ref == "GRCh38",
           region == "gene") %>% 
    select(-vars, -svs) %>% 
    pivot_wider(names_from = benchmark, values_from = cov) %>% 
    mutate(gene_cat = case_when(mrg > 0 ~ "MRG",
                                v4 > 0.9 ~ "V4",
                                v4 < 0.9 & hifi < 1 ~ "Not Benchmarked",
                                mrg == 0 & hifi == 1 ~ "Excluded?",
                                TRUE ~ "???"))

bench_cov_anno_df <- mrg_gene_cats %>% 
    left_join(gene_strat_df)
```

## Figure Panel
### testing modality
__TODO__ Update somatic and WES `since these are derivatives of lists, I wouldn't use these in this figure.  Instead, I'd just add Mandelker and COSMIC, but name Mandelker as "OMIM/HGMD/ClinVar"` (message from JZ on slack)
```{r}
mrg_info_tbl
```

```{r}
mrg_testing_modality_df <- mrg_gene_cats %>%
    filter(gene_cat == "MRG") %>%
    left_join(mrg_info_tbl) %>%
    mutate(
        mandelker = if_else(mandelker == "yes", 1, 0),
        COSMIC = if_else(cosmic == "yes", 1, 0)
    ) %>%
    
    pivot_longer(
        cols = c(
            "mandelker",
            "COSMIC",
            "ClinGen",
            "CPIC",
            "Counsyl",
            "ACMG",
            "HTD",
            "NCCN_ESMO",
            "ACOG"
        ),
        names_to = "testing_modality"
    ) %>%
    filter(value != 0) %>%
    mutate(
        testing_modality = if_else(
            testing_modality == "mandelker",
            "OMIM/HGMD/ClinVar",
            testing_modality
        )
    )

(testing_modality_upset_plt <- mrg_testing_modality_df %>% 
    group_by(ref, chrom, start, end, gene) %>%
    summarise(list_set = list(testing_modality)) %>% 
    ggplot(aes(x = list_set)) +
    geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.25) +
    scale_x_upset() + 
    theme_bw() +
    theme(plot.margin = margin(10, 10, 10, 60)) +
    labs(x = "Testing Modalities", y = "Number of Genes"))
```
```{r}
ggsave(here("figures","testing_modalities.png"), testing_modality_upset_plt, width = 6, height = 6)
```


### Difficult Coverage
```{r}
bench_cov_anno_df %>% 
    filter(ref == "GRCh38",
           gene_cat %in% c("V4", "MRG","Not Benchmarked"),
        str_detect(coords, "[MTXY]",negate = TRUE),
        strat == "allDiff") %>% 
    mutate(gene_cat = factor(gene_cat, 
                             level = c("V4", "MRG","Not Benchmarked"))
           ) %>% 
    ggplot() + 
    geom_histogram(aes(x = overlap, fill = gene_cat), 
                   color = "grey40", size = 0.25, bins = 100) + 
    facet_wrap(~gene_cat, ncol = 1, scales = "free_y") + 
    theme_bw() + 
    theme(legend.position = "bottom") + 
    labs(x = "Difficult Region and SV Overlap", 
         y = "Number of Genes",
         fill = "Benchmark")
```

## MRG Small Var Benchmark GATK
```{r}
tidy_gatk_bench_df <- gatk_bench_df %>% 
    select(-contains("PLUS"),
           -contains("het"),
           -contains("homalt")) %>% 
    mutate(SNP.fn_rate = 1 - SNP.Recall,
           INDEL.fn_rate = 1 - INDEL.Recall) %>% 
    pivot_longer(cols = -c(Query, Genome, Ref, benchmark, Subset), 
                 names_to = "metric") %>% 
    mutate(var_type = case_when(str_detect(metric, "INDEL") ~ "INDEL",
                                str_detect(metric, "SNP") ~ "SNV",
                                TRUE ~ "Other"),
           metric = str_remove(metric, "INDEL.|SNP.")) 
```

```{r}
library(ggrepel)
gatk_plt_df <- tidy_gatk_bench_df %>% 
    filter(metric %in% c("fn_rate", "Frac_NA")) %>% 
    select(benchmark, var_type, Subset, metric, value) %>% 
    pivot_wider(names_from = benchmark, values_from = value)

gatk_plt_df %>% 
    mutate(metric = if_else(metric == "fn_rate", "FN Rate", "Frac Not Assessed")) %>% 
    mutate(Subset = case_when(Subset == "*" ~ "MRG Benchmark Regions",
                              Subset == "GRCh38_MRG_benchmark_gene_coordinates_falselyduplicated.bed.gz" ~ "Falsely Duplicated Regions",
                              Subset == "GRCh38_notinalllowmapandsegdupregions.bed.gz" ~ "Not In LowMap and SegDup",
                              Subset == "HG002_GRCh38_MRG_stratification_large_duplication_not_in_ref.bed" ~ "HG002 Large Dups",
                              Subset == "GRCh38_MHC.bed.gz" ~ "MHC",
                              Subset == "GRCh38_notinalldifficultregions.bed.gz" ~ "Not In All Difficult",
                              Subset == "HG002_GRCh38_MRG_stratification_ComplexVar_in_TR.bed.gz" ~ "ComplexVars In TR",
                              TRUE ~ "DEBUGME")) %>% 
    ggplot(aes(x = v4.2.1, y = v0.03.00)) + 
    geom_abline(aes(intercept = 0, slope = 1), linetype = 2, color = "grey60") + 
    geom_text_repel(aes(label = Subset), size = 2) +
    geom_point(aes(fill = Subset), shape = 21, color = "grey60") + 
    facet_grid(var_type~metric) + 
    
    guides(fill = guide_legend(nrow = 3)) +
    scale_color_brewer(type = "qual", palette = 2) + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "V4.2 Benchmark", y = "MRG Benchmark", 
         shape = "Variant Type", fill = "Stratification")

```
## MRG Gene Coverage
```{r}
mrg_gene_cov <- bench_cov_anno_df %>% 
    filter(gene_cat == "MRG") %>% 
    select(gene, ref, coords) %>% 
    distinct() %>% 
    left_join(bench_asm_cov_df) %>% 
    filter(ref == "GRCh38", 
           str_detect(coords, "[MTXY]",negate = TRUE),
           benchmark == "mrg")

(mrg_gene_cov_plt <- ggplot(mrg_gene_cov) + 
    geom_histogram(aes(x = cov, fill = region), 
                   position = "dodge",
                   color = "grey40",
                   size = 0.25) + 
    theme_bw() +
    labs(x = "Benchmark Coverage", y = "Number of Genes", fill = "") +
    theme(legend.position = c(0.15,0.9), 
          legend.direction = "horizontal", legend.background = element_blank()))
```


# Extra
```{r}
# Load overlap for HG002 GRCh38 hifiasm v0.11 and 
#v4.2.1 of medically relevant gene coordinates
hifiasm_v11_and_v4.2.1_overlap_df_grch38 <- read_tsv(
    here("data","overlap_analysis", "GRCh38", "GRCh38_overlap_v4.2.1_hifiasm.tsv"),
    col_types = "cddcddddd"
)

GRCh38_medical_benchmark_genes <- read_tsv(
    here("data","mrg_benchmarkset", "HG002", "GRCh38", 
         "intermediate_files", "HG002_GRCh38_MRG_no_MHC.bed"),
    col_types = "cddc", col_names = c("chr", "start", "end", "gene")
)


mandelker_list <- read_tsv(
    here("data", "mrg_lists","Mandelker_Medically_Relevant_Genes.tsv"),
    col_types = "c")

cosmic_list <- read_tsv(here("data", "mrg_lists", "COSMIC_Gene_Census.tsv"),
                        col_types = "c")

lincoln_list <- read_tsv(
    here("data", "mrg_lists", "Steve_Lincoln_Compiled_Medical_Gene_List.tsv"),
    col_types = "c")

somatic_cancer <- setdiff(cosmic_list$Gene, union(mandelker_list$Gene, 
                                                  lincoln_list$Gene))


whole_exome_seq <- setdiff(mandelker_list$Gene, union(cosmic_list$Gene, 
                                                  lincoln_list$Gene))
    
clingen_mod_def_strong <- read_tsv(
    here("data", "mrg_lists", "ClinGen_moderate_definitive_strong.tsv"), 
    col_types = "c")

acmg_sf_2 <- read_tsv(here("data", 
                           "mrg_lists","ACMG_SF_2.0.tsv"), col_types = "c")

CPIC <- read_tsv(here("data", 
                      "mrg_lists","CPIC.tsv"), col_types = "c")

ACOG <- read_tsv(here("data", 
                      "mrg_lists","ACOG.tsv"), col_types = "c")

Counsyl <- read_tsv(here("data", 
                         "mrg_lists","Counsyl.tsv"), col_types = "c")

NCCN_ESMO <- read_tsv(here("data", 
                           "mrg_lists","NCCN_ESMO.tsv"), col_types = "c")

Assorted_HTD <- read_tsv(here("data", 
                              "mrg_lists","Assorted_HTD.tsv"), col_types = "c")

# Add columns for different levels of overlap 
med_gene_list_anno_df <- 
	hifiasm_v11_and_v4.2.1_overlap_df_grch38 %>% 
	mutate(overlap_v4 = ifelse(percent_overlap_v4.2.1 < .9, 
							   "v4_lt_90",
							   "v4_gte_90")) %>%
	mutate(asm_resolved = ifelse(percent_flanking_plus_segdups_overlap_hifiasm == 1 
								 & flanking_breaks_in_dip_bed == 0, 
								 "asm_pass",
								 "asm_fail")) %>%
    ## TODO - modify for in V4, in med genes, no in a benchmark
	mutate(benchmark_membership = case_when(
	    gene %in% GRCh38_medical_benchmark_genes$gene ~ "in_mrg",
	    overlap_v4 == "v4_gte_90" ~ "in_v4",
	    TRUE ~ "Not benchmarked")) %>%
    mutate(wes = if_else(gene %in% whole_exome_seq, "in WES", "not in WES")) %>% 
    mutate(test_modality = 
               case_when(gene %in% somatic_cancer ~ "Somatic_Cancer",
                         gene %in% clingen_mod_def_strong$Gene ~ "ClinGen",
                         gene %in% acmg_sf_2$Gene ~ "ACMG_SF_2",
                         gene %in% CPIC$Gene ~ "CPIC",
                         gene %in% ACOG$Gene ~ "ACOG",
                         gene %in% Counsyl$Gene ~ "Counsyl",
                         gene %in% NCCN_ESMO$Gene ~ "NCCN_ESMO",
                         gene %in% Assorted_HTD$Gene ~ "Assorted_HTD",
                         gene %in% whole_exome_seq ~ "WES",
                         TRUE ~ "None known"))

## Selecting columns for alluvial diagram
alluvial_df <- med_gene_list_anno_df %>% 
    select(gene,chrom, start, end, overlap_v4, asm_resolved, 
       benchmark_membership, wes, test_modality) 
```

## Benchmarking Medically Relevant Genes
__Questions__  
- 20 genes in mrg that are also in v4  
- 270 genes not benchmarked with fully dip assembled

```{r}
med_gene_list_anno_df %>%
    count(overlap_v4, asm_resolved, benchmark_membership) %>%
    pivot_wider(names_from = benchmark_membership,
                values_from = n,
                values_fill = 0)
```

```{r}
med_gene_list_anno_df %>% 
    filter(overlap_v4 == "v4_lt_90", 
           asm_resolved == "asm_fail",
           benchmark_membership == "in_mrg") %>% 
    select(gene, chrom, start, end)
```
```{r}
med_gene_list_anno_df %>% 
    filter(overlap_v4 == "v4_gte_90", 
           asm_resolved == "asm_pass",
           benchmark_membership == "in_mrg")
```

## Exploratory Figures
```{r}
alluvial_df %>% 
    ggplot() + 
    geom_bar(aes(x = benchmark_membership, fill = test_modality), 
             position = "fill") + 
    facet_grid(overlap_v4~asm_resolved) + 
    theme_bw()
```


```{r}
alluvial_df %>% 
    pivot_longer(cols = -c(gene, test_modality), 
                 names_to = "cat", values_to = "cat_value") %>% 
    ggplot() + 
    geom_bar(aes(x = cat_value, fill = test_modality)) + 
    facet_wrap(~cat, scales = "free") + 
    theme_bw()
```

```{r}
bench_counts <- med_gene_list_anno_df %>%  
    count(benchmark_membership) %>% 
    mutate(benchmark_membership = factor(
        benchmark_membership,
        levels = c("in_v4", "in_mrg", "Not benchmarked")
    )) 

med_gene_list_anno_df %>%
    mutate(benchmark_membership = factor(
        benchmark_membership,
        levels = c("in_v4", "in_mrg", "Not benchmarked")
    )) %>%
    ggplot() +
    geom_bar(aes(x = benchmark_membership, fill = test_modality),
             position = "dodge") +
    geom_text(data = bench_counts, 
              aes(x = benchmark_membership, y = 0.75, label = n)) + 
    theme_bw() +
    scale_fill_brewer(type = "qual", palette = 3) +
    scale_y_log10() + 
    theme(legend.position = "bottom")
```




# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```