---
title: "Medically Relevant Genes Characterizations"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
library(readxl)
library(ggupset)
library(ggrepel)
library(ggpubr)
```

## Medical Genes Benchmark
**Panels**

{DRAFT} A. Gene assembly coverage - flanking diagram  - Illustrator diagram
{DRAFT} B. Medical composition - Upset plot with testing modalities 
{DRAFT} C. Small variant benchmark (poster figure) over GATK  
{DRAFT} D. Coverage by gene part (exon, introns) for SV and small var union  
{DRAFT} E. Number of variants (SNV, indels, and SVs) in benchmark by gene part (exon, introns) for all and genes in high priority list
{DRAFT} E. small var sv and indel size distributions

**Key Points**\
- Identification of candidate genes\
- 274 genes with know significant relevance\
- Not all genes are benchmarkable - coverage of genes and number of
small and structural variants, bases / variants excluded due to
different filters.

# Loading and Tidying Data
```{r}
## Gene source and testing modality information
mrg_info_tbl <- here("data","tidy","gene_list_info.tsv") %>% 
    read_tsv(col_types = "cccddcccllddddddd")

## Gene Stratfication Overlap
gene_strat_df <- read_tsv(here("data","tidy","gene_strat_overlap.tsv"))

## Benchmark and Asm Coverage information
bench_asm_cov_df <- here("data","tidy","gene_asm_bench_coverage.tsv") %>% 
    read_tsv(col_types = "ccclccddd")

## MRG versus GATK
# Extracting Column Names
gatk_colnames <- here("data","benchmark_evaluation",
                      "smvar HG002 MRG v0.03.00 evaluation.xlsx") %>% 
    read_excel(sheet = "summary stats NYGC", range = "B1:AR2", na = "NA")

## Getting benchmark results
gatk_bench_df <- here("data",
                      "benchmark_evaluation",
                      "smvar HG002 MRG v0.03.00 evaluation.xlsx") %>%
    read_excel(
        sheet = "summary stats NYGC",
        range = "B32:AR45",
        na = "NA",
        col_names = colnames(gatk_colnames)
    )
```

Combining Benchmark Coverage and Strat data frames
```{r}
mrg_gene_cats <- bench_asm_cov_df %>% 
    filter(str_detect(coords, "[MTXY]",negate = TRUE),
           ref == "GRCh38",
           region == "gene") %>% 
    select(-vars, -svs) %>% 
    pivot_wider(names_from = benchmark, values_from = cov) %>% 
    mutate(gene_cat = case_when(in_mrg ~ "MRG",
                                v4 > 0.9 ~ "V4",
                                v4 < 0.9 & hifi < 1 ~ "Not Benchmarked",
                                mrg == 0 & hifi == 1 ~ "Excluded?",
                                TRUE ~ "???"))

bench_cov_anno_df <- mrg_gene_cats %>% 
    left_join(gene_strat_df)
```

## Figure Panel
### testing modality
__TODO__ Update somatic and WES `since these are derivatives of lists, I wouldn't use these in this figure.  Instead, I'd just add Mandelker and COSMIC, but name Mandelker as "OMIM/HGMD/ClinVar"` (message from JZ on slack)
```{r}
mrg_info_tbl
```

```{r}
mrg_testing_modality_df <- mrg_gene_cats %>%
    filter(gene_cat == "MRG") %>%
    left_join(mrg_info_tbl) %>%
    mutate(
        mandelker = if_else(mandelker == "yes", 1, 0),
        COSMIC = if_else(cosmic == "yes", 1, 0)
    ) %>%
    
    pivot_longer(
        cols = c(
            "mandelker",
            "COSMIC",
            "ClinGen",
            "CPIC",
            "Counsyl",
            "ACMG",
            "HTD",
            "NCCN",
            "ACOG"
        ),
        names_to = "testing_modality"
    ) %>%
    filter(value != 0) %>%
    mutate(
        testing_modality = if_else(
            testing_modality == "mandelker",
            "OMIM/HGMD/ClinVar",
            testing_modality
        )
    )

(testing_modality_upset_plt <- mrg_testing_modality_df %>% 
    group_by(ref, chrom, start, end, gene) %>%
    summarise(list_set = list(testing_modality)) %>% 
    ggplot(aes(x = list_set)) +
    geom_bar() +
    geom_text(stat='count', aes(label=after_stat(count)), vjust=-0.25) +
    scale_x_upset() + 
    theme_bw() +
    theme(plot.margin = margin(10, 10, 10, 60)) +
    labs(x = "Testing Modalities", y = "Number of Genes"))
```
```{r}
ggsave(here("figures","testing_modalities.png"), testing_modality_upset_plt, width = 6, height = 6)
```


### Difficult Coverage
```{r}
bench_cov_anno_df %>% 
    filter(ref == "GRCh38",
           gene_cat %in% c("V4", "MRG","Not Benchmarked"),
        str_detect(coords, "[MTXY]",negate = TRUE),
        strat == "allDiff") %>% 
    mutate(gene_cat = factor(gene_cat, 
                             level = c("V4", "MRG","Not Benchmarked"))
           ) %>% 
    ggplot() + 
    geom_histogram(aes(x = overlap, fill = gene_cat), 
                   color = "grey40", size = 0.25, bins = 100) + 
    facet_wrap(~gene_cat, ncol = 1, scales = "free_y") + 
    theme_bw() + 
    theme(legend.position = "bottom") + 
    labs(x = "Difficult Region and SV Overlap", 
         y = "Number of Genes",
         fill = "Benchmark")
```

## MRG Small Var Benchmark GATK
```{r}
tidy_gatk_bench_df <- gatk_bench_df %>% 
    select(-contains("PLUS"),
           -contains("het"),
           -contains("homalt")) %>% 
    mutate(SNP.fn_rate = 1 - SNP.Recall,
           INDEL.fn_rate = 1 - INDEL.Recall) %>% 
    pivot_longer(cols = -c(Query, Genome, Ref, benchmark, Subset), 
                 names_to = "metric") %>% 
    mutate(var_type = case_when(str_detect(metric, "INDEL") ~ "INDEL",
                                str_detect(metric, "SNP") ~ "SNV",
                                TRUE ~ "Other"),
           metric = str_remove(metric, "INDEL.|SNP.")) 
```

```{r}

gatk_plt_df <- tidy_gatk_bench_df %>% 
    filter(metric %in% c("fn_rate", "Frac_NA")) %>% 
    select(benchmark, var_type, Subset, metric, value) %>% 
    pivot_wider(names_from = benchmark, values_from = value)

gatk_plt_df %>% 
    mutate(metric = if_else(metric == "fn_rate", "FN Rate", "Frac Not Assessed")) %>% 
    mutate(Subset = case_when(Subset == "*" ~ "MRG Benchmark Regions",
                              Subset == "GRCh38_MRG_benchmark_gene_coordinates_falselyduplicated.bed.gz" ~ "Falsely Duplicated Regions",
                              Subset == "GRCh38_notinalllowmapandsegdupregions.bed.gz" ~ "Not In LowMap and SegDup",
                              Subset == "HG002_GRCh38_MRG_stratification_large_duplication_not_in_ref.bed" ~ "HG002 Large Dups",
                              Subset == "GRCh38_MHC.bed.gz" ~ "MHC",
                              Subset == "GRCh38_notinalldifficultregions.bed.gz" ~ "Not In All Difficult",
                              Subset == "HG002_GRCh38_MRG_stratification_ComplexVar_in_TR.bed.gz" ~ "ComplexVars In TR",
                              TRUE ~ "DEBUGME")) %>% 
    ggplot(aes(x = v4.2.1, y = v0.03.00)) + 
    geom_abline(aes(intercept = 0, slope = 1), linetype = 2, color = "grey60") + 
    geom_text_repel(aes(label = Subset), size = 2) +
    geom_point(aes(fill = Subset), shape = 21, color = "grey60") + 
    facet_grid(var_type~metric) + 
    
    guides(fill = guide_legend(nrow = 3)) +
    scale_color_brewer(type = "qual", palette = 2) + 
    theme_bw() + 
    theme(legend.position = "none") + 
    labs(x = "V4.2 Benchmark", y = "MRG Benchmark", 
         shape = "Variant Type", fill = "Stratification")

```
## MRG Gene Coverage
```{r}
(mrg_gene_cov_plt <- mrg_gene_cov %>% 
     filter(bench_type == "union") %>% 
     ggplot() + 
    geom_histogram(aes(x = cov, fill = region), 
                   position = "dodge",
                   color = "grey40",
                   size = 0.25, bins = 50) + 
    theme_bw() +
    labs(x = "Fraction included in combined small and structural variant benchmark", 
         y = "# of Genes", fill = "") +
    theme(legend.position = c(0.25,0.9), 
          legend.direction = "horizontal", 
          legend.background = element_blank()))
```
```{r}
ggsave(
    here("figures", "mrg_gene_coverage.png"),
    mrg_gene_cov_plt,
    width = 6,
    height = 4,
    dpi = 300
)
```

## Number of Variants by Gene Part

```{r}
hp_genes_df <- mrg_info_tbl %>% select(gene, high_priority) %>% distinct()

mrg_var_count_df <- bench_asm_cov_df %>% 
    left_join(hp_genes_df) %>% 
    filter(in_mrg, benchmark == "mrg", ref == "GRCh38") %>% 
    select(-in_mrg, -cov)

mrg_var_count_plt_df <- mrg_var_count_df %>%
    pivot_longer(
        cols = c("vars", "svs"),
        names_to = "var_type",
        values_to = "n_vars"
    ) %>% 
    mutate(var_type = case_when(var_type == "vars" ~ "Small Vars",
                                var_type == "svs" ~ "Structural Vars")) %>% 
    group_by(benchmark, region, high_priority, var_type) %>% 
    summarise(total_vars = sum(n_vars))

(mrg_var_count_plt <- mrg_var_count_plt_df %>% 
    ggplot() + 
    geom_bar(aes(x = high_priority, y = total_vars, fill = region), 
             stat = "identity") + 
    facet_wrap(~var_type, scales = "free", ncol  = 1) + 
    theme_bw() + 
    labs(fill = "Gene Region", 
         x = "Variant in HP Gene",
         y = "# of Variants") + 
    theme(legend.position = "bottom")) 
```


## INDEL Size Distribution
```{r}
mrg_var_df <- read_tsv(here("data","tidy","mrg_var_tbl.tsv"),
                       col_types = "cciccccici")

(smv_size_dist_plt <- mrg_var_df %>% 
    filter(TYPE == "INDEL", bench_type == "smvar") %>% 
    mutate(gene_part = if_else(EXON == ".", "intro","exon")) %>% 
    ggplot() + 
    geom_bar(aes(x = var_size, fill = gene_part)) + 
        scale_fill_brewer(type = "qual", palette = 3) +
    theme_bw() +
    labs(x = "Small INDEL Size (bp)", y = "# of Variants", fill = "Gene Part") + 
    theme(legend.position = c(0.15, 0.8), legend.background = element_blank()))
```

```{r}
breaks <- c(50, 100, 500, 1000, 5000, 10000)
log_breaks <- log10(breaks)

sv_log_breaks <- sort(c(-log_breaks + log10(25), 0,  log_breaks - log10(25)))

sv_breaks <- sort(c(-breaks, 0, breaks))

(sv_size_dist_plt <- mrg_var_df %>% 
    filter(TYPE == "INDEL", bench_type == "sv") %>% 
    mutate(gene_part = if_else(EXON == ".", "intro","exon")) %>% 
    ## Log scale var length with 1 offset 
    mutate(var_size = if_else(var_size < 0, 
                              -log10(abs(var_size)) + log10(25),
                              log10(var_size) - log10(25))) %>% 
    ggplot() + 
    geom_histogram(aes(x = var_size, fill = gene_part), 
                   color = "grey40", size = 0.25, binwidth = 0.125) +
    scale_x_continuous(breaks = sv_log_breaks,
                       labels = sv_breaks) + 
        scale_fill_brewer(type = "qual", palette = 3) + 
    theme_bw() +
    theme(axis.text.x = element_text(angle = -45, hjust = 0),
          legend.position = c(0.15, 0.8), legend.background = element_blank()) + 
    labs(x = "Large INDEL Size (bp)", y = "# of Variants"))
```

## Combined Plot
```{r}
ggarrange(
    testing_modality_upset_plt,
    mrg_gene_cov_plt,
    ggarrange(
        mrg_var_count_plt + rremove("legend"),
        smv_size_dist_plt,
        sv_size_dist_plt + rremove("legend"),
        nrow = 1,
        widths = c(1, 2, 2), 
        labels = c("C","D", "E")
    ),
    labels = c("A","B"),
    heights = c(2,1,2),
    nrow = 3
)
```

```{r}
ggsave(here("figures","mrg_characterization.png"), 
       width = 8, height = 12, dpi = 300)
```





# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```