---
title: "Making Tidy Dataset for MRG Manuscript"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(readxl)
library(here)
```

__TODOs__
-[] Add data descriptors

# Objective
Generate tidy data sets for generating manuscript figures and tables.

## Datasets

1. Medical Gene List (`data/tidy/mrg_list_info.tsv`): gene coordinates, gene list membership, and testing modality  
1. Genes coverage (`data/tidy/gene_asm_bench_coverage.tsv`): benchmark set and asm coverage information  
1. Gene stratification overlap (`data/tidy/gene_strat_overlap.tsv`): overlap fraction of difficult stratifications with medically relevant genes.
1. Sequencing read coverage (`data/tidy/seq_coverage.tsv`): Gene Coverage for by different datasets


# Making Tidy Data Tables
Loading summary stat table generated by JW
```{r}
tbl_list <- list(
  GRCh37 = here("data", "gene_stat_tbls", "MRG_stats_table_GRCh37.tsv"),
  GRCh38 = here("data", "gene_stat_tbls", "MRG_stats_table_GRCh38.tsv")
)

c_types <- str_c("cddddc",
                 str_c(rep("d", 36), collapse = ""),
                 "ccc",
                 str_c(rep("d", 16), collapse = ""))

GRCh37_tbl <- read_tsv(tbl_list$GRCh37, col_types = c_types) %>%
  mutate(chrom = as.character(chrom)) %>% 
  rename(bp_hifiasm_dip.bed = "bp_coordinates_plus_segdups_plus_flanking_hifiasm_dip.bed",
         fraction_hifiasm_dip.bed = "fraction_coordinates_plus_segdups_plus_flanking_hifiasm_dip.bed")



c_types <- str_c("ccccddddc", str_c(rep("d", 51), collapse = ""))
GRCh38_tbl <- read_tsv(tbl_list$GRCh38, col_types = c_types)



mrg_stats_tbl <- bind_rows(GRCh37 = GRCh37_tbl, 
                           GRCh38 = GRCh38_tbl, 
                           .id = "ref")
```

```{r}
glimpse(mrg_stats_tbl)
```



## Medical Gene List
Extracting gene source information from main MRG table
```{r}
mrg_source_tbl <- mrg_stats_tbl %>% 
    select(gene, ref, chrom, start, end, 
           mandelker, cosmic, high_priority)
```


_DATA CHECK:_ Check for genes with multiple entries
__QUESTION:__ Determine reason for duplicate entries, are they table errors
33 genes with duplicate coordiante entries Using gene coordinates to differentiate.
```{r}
mrg_source_tbl %>%
  group_by(gene, ref) %>%
  mutate(count = n()) %>%
  filter(count != 1) %>%
  arrange(gene)
```

_DATA CHECK:_ Verifying only one entry per gene + coordinate combination. Should be 0 rows
```{r}
mrg_source_tbl %>% 
    count(gene, ref, chrom, start, end) %>% 
    filter(n > 1)
```

__QUESTION:__ Is ClinGen a testing modality?
__QUESTION:__ Is the test modality information ref genome specific at all? (potentially a stupid question)

__TODO:__ Get testing modality source info

Information on testing modality and high priority gene list from Lincoln https://drive.google.com/drive/u/1/folders/1gUjMx24CkZCbt4vC46W3nQY_3DcjP4X9 


Gene testing modality from individual list generated by Justin Wagner in collaboration with Steve Lincoln.
```{r}
testing_modality_df <- list(ACMG = "ACMG_SF_2.0.tsv",
                             ACOG = "ACOG.tsv",
                             HTD = "Assorted_HTD.tsv",
                             ClinGen = "ClinGen_moderate_definitive_strong.tsv",
                             Counsyl = "Counsyl.tsv",
                             CPIC = "CPIC.tsv",
                             NCCN_ESMO = "NCCN_ESMO.tsv") %>% 
    map(~here("data","mrg_lists","testing_modalities", .)) %>% 
    map_df(read_tsv, col_types = "c", .id = "testing_modality") %>% 
    count(Gene, testing_modality) %>% 
    pivot_wider(names_from = testing_modality,
                values_from = n,
                values_fill = 0) %>% 
    rename(gene = "Gene")
```

### Annotating with Somatic and WES testing modalities

__QUESTION__ how are these defined, i.e. how do we know these set operations return the correct gene set.
Code from JW MRG_alluvial_diagram script


```
somatic_cancer <- setdiff(cosmic_list$Gene, union(mandelker_list$Gene, 
                                                  lincoln_list$Gene))


whole_exome_seq <- setdiff(mandelker_list$Gene, union(cosmic_list$Gene, 
                                                  lincoln_list$Gene))
```
                                                  
```{r}
mrg_info_tbl <- mrg_source_tbl %>%
    mutate(wes = (mandelker == "no" &
                      (cosmic == "yes" | high_priority == "yes")),
           somatic = (cosmic == "no" &
                          (mandelker == "yes" | high_priority == "yes"))) %>% 
    full_join(testing_modality_df)
```

_DATA CHECK:_ Checking that all genes in testing modality list are in mrg source list
All genes in the testing modality list are in the combined GRCh37 and GRCh38 gene lists. 

```{r}
nrow(mrg_info_tbl) == nrow(mrg_source_tbl)
```

Differing numbers of genes with and without testing modalities for the two reference genomes. 
This might be expected. 
Related to question above about which ref genome the testing modalities are relevant to.
```{r}
mrg_info_tbl %>% 
  mutate(test_modality_info = !is.na(ClinGen)) %>% 
  count(ref, test_modality_info)
```


_DATA CHECK:_ Verifying WES and Somatic testing modality correctly assigned
```{r}
mrg_info_tbl %>% 
    count(mandelker, cosmic, high_priority, wes)
```
```{r}
mrg_info_tbl %>% 
    count(mandelker, cosmic, high_priority, somatic)
```

Tidying tibble up for export
```{r}
tidy_mrg_info_tbl <- mrg_info_tbl %>%
    ## replacing NAs from testing modality with 0s
    mutate(across(where(is.integer), replace_na, 0))
```

### Example Gene
```{r}
tidy_mrg_info_tbl %>%
  filter(gene == "OR4F5") %>%
  DT::datatable()
```

### Writing To File 
```{r}
write_tsv(tidy_mrg_info_tbl,
          here("data","tidy","gene_list_info.tsv"))
```


## Benchmark and Assembly Coverage
Generating tidy table with gene and exon coverage for the medically relevant genes benchmark, v4 small variant benchmark, and hifi assembly.



__QUESTION__ Are these 440 NA values expected for GRCh38
Number of entries with NA values for MRG v0.03 and SV 0.02
```{r}
mrg_stats_tbl %>% 
  group_by(ref) %>% 
  summarise(across(where(is.numeric), ~ sum(is.na(.x)))) %>% 
  pivot_longer(cols = -ref) %>% 
  group_by(name) %>% 
  mutate(total = sum(value)) %>% 
  filter(total != 0) %>% 
  arrange(-total, name) %>% 
  select(-total)
```

```{r}
na_genes <- mrg_stats_tbl %>% 
  filter(is.na(fraction_exon_covered_MRG_small_variant_v0.03_benchmark)) %>% 
  {.$gene} %>% unique()
mrg_stats_tbl %>% 
  filter(gene %in% na_genes,
         str_detect(chrom, "[MTXY]", negate = TRUE)) %>% 
  arrange(gene)
```


```{r}
mrg_bench_asm_df <- mrg_stats_tbl %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end)) %>% 
    select(gene, ref, coords, 
           contains("V4"), 
           contains("MRG"), 
           -starts_with("bp_"), 
           -contains("CNV"), 
           -contains("stratification"),
           fraction_hifiasm_dip.bed,
          ## excluding small MRG v0.2 var counts and coverage
          -fraction_MRG_benchmark_bed,
          -fraction_exon_covered_MRG_benchmark,
           -num_MRG_small_variant_v0.2_variants_in_gene,
           -num_MRG_small_variant_v0.2_variants_in_exons,
          ## excluding SVs MRG v0.1 counts
          -num_MRG_SV_v0.1_in_gene,
          -num_MRG_SV_v0.1_benchmark_variants_in_exons) %>% 
  ## Renaming columns for consistent formatting
    rename(v4_gene_cov = fraction_HG002_v4.2.1,
           v4_exon_cov = fraction_exon_covered_v4.2.1,
           v4_gene_vars = num_v4.2.1_small_variants_in_gene,
           v4_exon_vars = num_v4.2.1_variants_in_exons,
           mrg_gene_cov = fraction_MRG_small_variant_v0.03,
           mrg_exon_cov = fraction_exon_covered_MRG_small_variant_v0.03_benchmark,
           mrg_gene_vars = num_MRG_small_variant_v0.03_variants_in_gene,
           mrg_exon_vars = num_MRG_small_variant_v0.03_variants_in_exons,
           mrg_gene_svs = num_MRG_SV_v0.02_SVs_in_gene,
           mrg_exon_svs = num_MRG_SV_v0.02_variants_in_exons,
           hifi_gene_cov = fraction_hifiasm_dip.bed) %>% 
    pivot_longer(cols = -c("gene", "ref", "coords"),
                 names_to = c("benchmark", "region","metric"),
                 names_sep = "_", values_to = "value") %>%
    pivot_wider(id_cols = c(gene, ref, coords, benchmark, region),
                names_from = metric, values_from = value)
```


### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_bench_asm_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable(rownames = FALSE)
```

### Writing To File 
```{r}
bench_cov_file <-  here("data","tidy","gene_asm_bench_coverage.tsv")

write_tsv(mrg_bench_asm_df, bench_cov_file)
```

## Stratification Coverage
```{r}
colnames(mrg_stats_tbl)
```

```{r}
mrg_strat_df <- mrg_stats_tbl %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end)) %>%  
    select(ref, gene, coords, 
           contains("fraction"),
           -contains("V4.2"), 
           -contains("MRG"), 
           -fraction_hifiasm_dip.bed) %>% 
    rename(giabSV = fraction_in_Tier1andTier2_SV_0.6_slop150_stratification,
           repeatsGt10kb = fraction_in_AllRepeats_gt_10kb,
           asmCNV = fraction_in_assembly_cnv_stratification,
           vdj = fraction_in_vdj_stratification,
           inversion = fraction_in_inversions_stratification,
           segdupHiCntHiPct = fraction_in_segmental_duplications_gt5_percent_identity_gte_990_stratification,
           segdup = fraction_in_segmental_duplications_stratification,
           lowmapAndSegDup = fraction_in_low_mappability_and_segmental_duplications_stratification,
           intersectCNV =fraction_in_illumina_ccs_ont_cnv_stratification,
           outlierCNV = fraction_in_ccs_ont_elliptical_outlier_cnv_stratification,
           allDiff = fraction_in_HG002_GIABv4.1_complexandSVs_alldifficultregions_stratification,
           refGaps = fraction_in_reference_gaps_stratification,
           CNVandSVs = fraction_in_HG002_GIABv4.1_CNVsandSVs_stratification) %>% 
    pivot_longer(cols = -c("ref", "gene", "coords"), 
                 names_to = "strat", values_to = "overlap")
```

```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_strat_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable()
```

### Writing Gene List Table to file 
```{r}
write_tsv(mrg_strat_df,
          here("data","tidy","gene_strat_overlap.tsv"))
```


## Sequencing Read Coverage Data
__TODO__ Get better input source data
Only coverage information for GRCh37, file downloaded from `https://gitlab.nist.gov/gitlab/jmw7/medically_relevant_characterization/-/raw/master/GRCh37/GRCh37_characterizations_assembly_contig_coverages_merged.tsv`
Manually modified file, only keeping gene list and read coverage columns.

```{r}
grch37_gene_cov_df <- read_tsv(
  here(
    "data",
    "seq_coverage",
    "GRCh37_characterizations_assembly_contig_coverages_merged.tsv"
  )
)

gene_cov_df <- grch37_gene_cov_df %>% 
  rename(chrom = Chromosome,
         gene = `Gene Symbol`) %>% 
  mutate(start = `Start minus 20kbp` + 20000,
         end = `End plus 20kbp` - 20000,
         ref = "GRCh37") %>% 
  select(-`Start minus 20kbp`, -`End plus 20kbp`)
```


_DATA Check:_ Verifying genes and coordinates match other data sources
```{r}
gene_check_df <- gene_cov_df %>% 
  mutate(chrom = paste0("chr",chrom)) %>% 
  full_join(mrg_stats_tbl)
```
```{r}
## Gene lists not matching as they are for different reference versions
nrow(gene_check_df)
```

```{r}
gene_check_df %>% arrange(gene) %>% 
  select(GRCh37, chrom, gene, start, end) %>% 
  top_n(10)
```


### Tidying data
```{r}
tidy_gene_cov_df <- gene_cov_df %>%
  select(gene, ref, chrom, start, end, contains("cover")) %>%
  gather(key = "analysis",
         value = "value",
         -gene,
         -ref,
         -chrom,
         -start,
         -end) %>%
  filter(!is.na(value)) %>%
  mutate(
    data_set = case_when(
      str_detect(analysis, "CCS") ~ "CCS",
      str_detect(analysis, "10X") ~ "10X",
      str_detect(analysis, "ONT") ~ "ONT"
    ),
    cov_stat = if_else(
      str_detect(analysis, "average"),
      "avg_cov",
      str_extract(analysis, "(?<=covered.by.).*(?=.CCS|.ONT|.10X)")
    )
  ) %>%
  mutate(cov_stat = str_replace(cov_stat, "at.least.", "bp_cov_by_")) %>%
  select(-analysis) %>%
  spread(cov_stat, value)
```


### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
tidy_gene_cov_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable()
```

### Writing To File 
```{r}
write_tsv(tidy_gene_cov_df,
          here("data","tidy","seq_coverage.tsv"))
```

## Sanger Confirmation
Summary table at https://docs.google.com/spreadsheets/d/1_29bJkQ1snrKb9TGeCWh2hRl6Rx7tPVd/edit#gid=985369016


## Evaluation / Manual Curation
_Note from Slack_ RE: spreadsheet source 
    @nolson for the evaluation results figure, I think you'll want to use my curation results in columns T-U, unless they are blank, in which case you should use columns Q-S, in the GRCh37andGRCh38 tab in https://docs.google.com/spreadsheets/d/1Pn7WP78JfWKCO2Df31n_4gzDOwtP69flgDyyjeBS6JE/edit?usp=sharing.  This'll be a bit more complicated than previous figs, since we have the CommonFP/FN/FP_FN rows, so happy to chat about this when you get to it (edited) 
    

    
### Loading and Tiydying
```{r}
eval_sheet <- "combined curation responses from benchmarking with sm variant v0.02.03.xlsx"
raw_bench_eval_df <- read_excel(here("data", "benchmark_evaluation", eval_sheet), 
                            sheet = "GRCh37andGRCh38", 
                            .name_repair = "universal") 
```
```{r}
glimpse(raw_bench_eval_df)
```

```{r}
tidy_bench_eval_df <- raw_bench_eval_df %>%
  rename(
    jz_benchmark_correct = "JZ..Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.benchmark...yes..no..or.unsure.",
    jz_query_correct = "JZ...Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.query.callset...yes..no..or.unsure.",
    benchmark_correct = "Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.benchmark...yes..no..or.unsure.",
    query_correct = "Is.this.variant.and.all.variants.and.genotypes.in.the.surrounding.region..e.g...in.the.same.tandem.repeat..correct.in.the.query.callset...yes..no..or.unsure.",
    fp_or_fn = "Is.this.a.putative.FP..FN.or.genotype.error..FP_FN..in.your.calls.",
    truth_gt = "truth_gt.or.for..common..truth_gt.when.comparing.to.Roche.callset",
    query_gt = "query_gt.or.for..common..query_gt.when.comparing.to.Roche.callset",
    ref_base = ref,
    alt_base = alt
  ) %>%
  select(
    callset.ID,
    reference,
    SNP.or.INDEL,
    chrwithprefix,
    GRCh37_POS...8,
    GRCh38_POS,
    fp_or_fn,
    truth_gt,
    query_gt,
    ref_base,
    alt_base,
    jz_benchmark_correct,
    jz_query_correct,
    benchmark_correct,
    query_correct
  ) %>%
  ## Replacing benchmark_correct and query_correct with JZ responses
  mutate(
    benchmark_correct = if_else(
      is.na(jz_benchmark_correct),
      benchmark_correct,
      jz_benchmark_correct
    ),
    query_correct = if_else(is.na(jz_query_correct), query_correct, jz_query_correct)
  ) %>%
  select(-jz_benchmark_correct,-jz_query_correct) %>% 
  rename(ref = reference,
         callset = callset.ID,
         var_type = SNP.or.INDEL,
         GRCh37_POS = GRCh37_POS...8) %>% 
  mutate(benchmark_correct = tolower(benchmark_correct),
         query_correct = tolower(query_correct))
```

### Writing to file
```{r}
write_tsv(tidy_bench_eval_df, 
          here("data","tidy","manual_curation.tsv"))
```


# Getting MD5s
```{r}
list.files(here("data","tidy")) %>% 
  {tools::md5sum(here("data","tidy", .))}
```


# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```