---
title: "Making Tidy Dataset for MRG Manuscript"
date: '`r Sys.Date()`'
output: 
    bookdown::html_document2:
        toc: true
        toc_float: true
        df_print: paged
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
```

__TODOs__
-[] Add data descriptors

# Objective
Generate tidy data sets for generating manuscript figures and tables.

## Datasets

1. Medical Gene List (`data/tidy/mrg_list_info.tsv`): gene coordinates, gene list membership, and testing modality  
1. Genes coverage (`data/tidy/gene_asm_bench_coverage.tsv`): benchmark set and asm coverage information  
1. Gene stratification overlap (`data/tidy/gene_strat_overlap.tsv`): overlap fraction of difficult stratifications with medically relevant genes.
1. Sequencing read coverage (`data/tidy/seq_coverage.tsv`): Gene Coverage for by different datasets

# Making Tidy Data Tables
Starting with list for GRCh38??? provided by JW
__TODO:__ incorporate ref genome version
```{r}
## Note currently in scratch ...
mrg_stats_tbl <- read_tsv(here("scratch","MRG_stats_table.tsv"))
```

## Medical Gene List
Extracting gene source information from main MRG table
```{r}
mrg_source_tbl <- mrg_stats_tbl %>% 
    select(gene, chrom, start, end, 
           mandelker, cosmic, high_priority)
```


_DATA CHECK:_ Check for genes with multiple entries
__QUESTION:__ Determine reason for duplicate entries, are they table errors
Seven Genes with duplicate entries. Using gene coordinates to differentiate.
```{r}
mrg_source_tbl %>% 
    group_by(gene) %>% 
    mutate(count = n()) %>% 
    filter(count != 1)
```

_DATA CHECK:_ Verifying only one entry per gene + coordinate combination. Should be 0 rows
```{r}
mrg_source_tbl %>% 
    count(gene, chrom, start,end) %>% 
    filter(n > 1)
```

__QUESTION:__ Is ClinGen a testing modality?
__TODO:__ Get testing modality source info
Information on testing modality and high priority gene list from Lincoln https://drive.google.com/drive/u/1/folders/1gUjMx24CkZCbt4vC46W3nQY_3DcjP4X9 


Gene testing modality from individual list generated by Justin Wagner in collaboration with Steve Lincoln.
```{r}
testing_modality_df <- list(ACMG = "ACMG_SF_2.0.tsv",
                             ACOG = "ACOG.tsv",
                             HTD = "Assorted_HTD.tsv",
                             ClinGen = "ClinGen_moderate_definitive_strong.tsv",
                             Counsyl = "Counsyl.tsv",
                             CPIC = "CPIC.tsv",
                             NCCN_ESMO = "NCCN_ESMO.tsv") %>% 
    map(~here("data","mrg_lists","testing_modalities", .)) %>% 
    map_df(read_tsv, col_types = "c", .id = "testing_modality") %>% 
    count(Gene, testing_modality) %>% 
    pivot_wider(names_from = testing_modality,
                values_from = n,
                values_fill = 0) %>% 
    rename(gene = "Gene")
```

### Annotating with Somatic and WES testing modalities

__QUESTION__ how are these defined, i.e. how do we know these set operations return the correct gene set.
Code from JW MRG_alluvial_diagram script


```
somatic_cancer <- setdiff(cosmic_list$Gene, union(mandelker_list$Gene, 
                                                  lincoln_list$Gene))


whole_exome_seq <- setdiff(mandelker_list$Gene, union(cosmic_list$Gene, 
                                                  lincoln_list$Gene))
```
                                                  
```{r}
mrg_info_tbl <- mrg_source_tbl %>%
    mutate(wes = (mandelker == "no" &
                      (cosmic == "yes" | high_priority == "yes")),
           somatic = (cosmic == "no" &
                          (mandelker == "yes" | high_priority == "yes"))) %>% 
    full_join(testing_modality_df)
```

_DATA CHECK:_ Checking that all genes in testing modality list are in mrg source list

- The testing modality annotated table has three more rows than the gene source table
```{r}
nrow(mrg_info_tbl)
nrow(mrg_source_tbl)
```

__QUESTION:__ Do we need to add these genes to the full gene list? Are they in other reference version
```{r}
mrg_info_tbl %>% filter(is.na(chrom))
```
```{r}
mrg_source_tbl %>% filter(str_detect(gene, "^IKB"))
```
```{r}
mrg_source_tbl %>% filter(str_detect(gene, "KAP"))
```


```{r}
mrg_source_tbl %>% filter(str_detect(gene, "^YA"))
```

```{r}
mrg_source_tbl %>% filter(str_detect(gene, "^MU"))
```


_DATA CHECK:_ Verifying WES and Somatic testing modality correctly assigned
```{r}
mrg_info_tbl %>% 
    count(mandelker, cosmic, high_priority, wes)
```
```{r}
mrg_info_tbl %>% 
    count(mandelker, cosmic, high_priority, somatic)
```

Tidying tibble up for export
```{r}
tidy_mrg_info_tbl <- mrg_info_tbl %>%
    ## Removing genes with testing modality but not in gene list
    filter(!is.na(chrom)) %>%
    ## replacing NAs from testing modality with 0s
    mutate(across(where(is.integer), replace_na, 0))
```

### Example Gene
```{r}
DT::datatable(tidy_mrg_info_tbl)
```

### Writing To File 
```{r}
write_tsv(tidy_mrg_info_tbl,
          here("data","tidy","gene_list_info.tsv"))
```


## Benchmark and Assembly Coverage
Generating tidy table with gene and exon coverage for the medically relevant genes benchmark, v4 small variant benchmark, and hifi assembly.

```{r}
wide_mrg_bench_asm_df <- mrg_stats_tbl %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end)) %>% 
    select(gene, coords, contains("V4"), contains("MRG"), 
           -starts_with("bp_"), -contains("CNV"), fraction_dip.bed) %>% 
    rename(v4_gene_cov = fraction_HG002_v4.2.1,
           v4_exon_cov = fraction_exon_covered_v4.2.1,
           v4_gene_vars = num_v4.2.1_variants_in_benchmark_in_gene,
           v4_exon_vars = num_v4.2.1_variants_in_benchmark_in_gene_in_exons,
           mrg_gene_cov = fraction_MRG_benchmark_bed,
           mrg_exon_cov = fraction_exon_covered_MRG_benchmark,
           mrg_gene_vars = num_MRG_variants_in_benchmark_in_gene,
           mrg_exon_vars = num_MRG_variants_in_benchmark_in_gene_in_exons,
           mrg_gene_svs = num_MRG_SVs_gt_49bp_in_SV_benchmark_bed_in_gene,
           mrg_exon_svs = num_MRG_SVs_gt_49bp_in_SV_benchmark_bed_in_gene_in_exons,
           hifi_gene_cov = fraction_dip.bed)

mrg_bench_asm_df <- wide_mrg_bench_asm_df %>% 
    pivot_longer(cols = -c("gene", "coords"), 
                 names_to = c("benchmark", "region","metric"), 
                 names_sep = "_", values_to = "value") %>% 
    pivot_wider(id_cols = c(gene, coords, benchmark, region),
                names_from = metric, values_from = value)
```


### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_bench_asm_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable()
```

### Writing To File 
```{r}
bench_cov_file <-  here("data","tidy","gene_asm_bench_coverage.tsv")
write_tsv(mrg_bench_asm_df, bench_cov_file)

```

## Stratification Coverage
```{r}
mrg_strat_df <- mrg_stats_tbl %>% 
    mutate(coords = str_c(chrom, ":",start,"-",end)) %>%  
    select(gene, coords, contains("fraction"), -contains("V4"), -contains("MRG"), 
           -fraction_dip.bed, 
           fraction_overlap_w_intersect_Illumina_CCS_ONT_CNV_v4.2.1, 
           fraction_overlap_w_union_CCS_and_ONT_outlier_CNV_v4.2.1) %>% 
    rename(refn = fraction_gene_overlap_w_refn,
           giabSV = fraction_overlap_w_GIAB_expanded_by_150_Tier1andTier2_SV_0.6,
           repeatsGt10kb = fraction_gene_overlap_w_AllRepeats_gt_10kb,
           asmCNV = fraction_overlap_w_Assembly_CNV,
           vdj = fraction_overlap_w_vdj,
           inversion = fraction_overlap_inversions,
           segdupHiCntHiPct = fraction_overlap_w_segdups_w_count_gt_5_and_percent_identity_gte_990,
           segdup = fraction_segdups,
           lowmapAndSegDup = fraction_lowmapandsegdups,
           intersectCNV = fraction_overlap_w_intersect_Illumina_CCS_ONT_CNV_v4.2.1,
           outlierCNV = fraction_overlap_w_union_CCS_and_ONT_outlier_CNV_v4.2.1) %>% 
    pivot_longer(cols = -c("gene", "coords"), 
                 names_to = "strat", values_to = "value")
```

```{r}
## Only displaying info for 1 gene to reduce html file size
mrg_strat_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable()
```

### Writing Gene List Table to file 
```{r}
write_tsv(mrg_strat_df,
          here("data","tidy","gene_strat_overlap.tsv"))
```


## Sequencing Read Coverage Data
__TODO__ Get better input source data
Only coverage information for GRCh37, file downloaded from `https://gitlab.nist.gov/gitlab/jmw7/medically_relevant_characterization/-/raw/master/GRCh37/GRCh37_characterizations_assembly_contig_coverages_merged.tsv`
Manually modified file, only keeping gene list and read coverage columns.

```{r}
grch37_gene_cov_df <- read_tsv(
  here(
    "data",
    "seq_coverage",
    "GRCh37_characterizations_assembly_contig_coverages_merged.tsv"
  )
)

gene_cov_df <- grch37_gene_cov_df %>% 
  rename(chrom = Chromosome,
         gene = `Gene Symbol`) %>% 
  mutate(start = `Start minus 20kbp` + 20000,
         end = `End plus 20kbp` - 20000,
         ref = "GRCh37") %>% 
  select(-`Start minus 20kbp`, -`End plus 20kbp`)
```


_DATA Check:_ Verifying genes and coordinates match other data sources
```{r}
gene_check_df <- gene_cov_df %>% 
  mutate(chrom = paste0("chr",chrom)) %>% 
  full_join(mrg_stats_tbl)
```
```{r}
## Gene lists not matching as they are for different reference versions
nrow(gene_check_df)
```

```{r}
gene_check_df %>% arrange(gene) %>% 
  select(GRCh37, chrom, gene, start, end) %>% 
  top_n(10)
```


### Tidying data
```{r}
tidy_gene_cov_df <- gene_cov_df %>%
  select(gene, ref, chrom, start, end, contains("cover")) %>%
  gather(key = "analysis",
         value = "value",
         -gene,
         -ref,
         -chrom,
         -start,
         -end) %>%
  filter(!is.na(value)) %>%
  mutate(
    data_set = case_when(
      str_detect(analysis, "CCS") ~ "CCS",
      str_detect(analysis, "10X") ~ "10X",
      str_detect(analysis, "ONT") ~ "ONT"
    ),
    cov_stat = if_else(
      str_detect(analysis, "average"),
      "avg_cov",
      str_extract(analysis, "(?<=covered.by.).*(?=.CCS|.ONT|.10X)")
    )
  ) %>%
  mutate(cov_stat = str_replace(cov_stat, "at.least.", "bp_cov_by_")) %>%
  select(-analysis) %>%
  spread(cov_stat, value)
```


### Example Gene
```{r}
## Only displaying info for 1 gene to reduce html file size
tidy_gene_cov_df %>% 
    filter(gene == "OR4F5") %>% 
    DT::datatable()
```

### Writing To File 
```{r}
write_tsv(tidy_gene_cov_df,
          here("data","tidy","seq_coverage.tsv"))
```

## Sanger Confirmation
Summary table at https://docs.google.com/spreadsheets/d/1_29bJkQ1snrKb9TGeCWh2hRl6Rx7tPVd/edit#gid=985369016


# Getting MD5s
```{r}
list.files(here("data","tidy")) %>% 
  {tools::md5sum(here("data","tidy", .))}
```


# Session Information
## System Information
```{r}
sessioninfo::platform_info()
```


## Package Versions
```{r}
sessioninfo::package_info() %>% 
    filter(attached = TRUE) %>% 
    select(package, loadedversion, date, source) %>%
    knitr::kable(booktabs = TRUE, row.names = FALSE)
```